<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Function reference â€¢ camprotR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-index">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">camprotR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/LFQ_parse_qc_protein_inference.html">LFQ QC and summarisation to protein-level abundance</a>
    </li>
    <li>
      <a href="../articles/SILAC.html">Processing and QC of SILAC data</a>
    </li>
    <li>
      <a href="../articles/TMT_PSM_QC_Summarisation.html">TMT QC PSM-level quantification and summarisation to protein-level abundance</a>
    </li>
    <li>
      <a href="../articles/crap.html">cRAP databases</a>
    </li>
    <li>
      <a href="../articles/msnset.html">Working with MSnSets</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index"><colgroup><col class="alias"><col class="title"></colgroup><tbody><tr><th colspan="2">
          <h2 id="all-functions">All functions <a href="#all-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="add_PTM_positions.html">add_PTM_positions()</a></code> </p>
        </td>
        <td><p>Add a column describing the position(s) of the PTM(s) with respect to the protein</p></td>
      </tr><tr><td>
          <p><code><a href="add_peptide_positions.html">add_peptide_positions()</a></code> </p>
        </td>
        <td><p>Add a column describing the position(s) of the peptide sequence with respect
to the protein</p></td>
      </tr><tr><td>
          <p><code><a href="add_site_sequence.html">add_site_sequence()</a></code> </p>
        </td>
        <td><p>Add a column with amino acid sequence around a PTM</p></td>
      </tr><tr><td>
          <p><code><a href="append_fasta.html">append_fasta()</a></code> </p>
        </td>
        <td><p>Append sequences to end of a FASTA</p></td>
      </tr><tr><td>
          <p><code><a href="center_normalise_to_ref.html">center_normalise_to_ref()</a></code> </p>
        </td>
        <td><p>Center-median normalise the expression matrix in an MSnSet using medians
from a reference dataset</p></td>
      </tr><tr><td>
          <p><code><a href="check_uniprot_release.html">check_uniprot_release()</a></code> </p>
        </td>
        <td><p>Check the current UniProt release</p></td>
      </tr><tr><td>
          <p><code><a href="count_features_per_protein.html">count_features_per_protein()</a></code> </p>
        </td>
        <td><p>Count the number of features (rows) per protein</p></td>
      </tr><tr><td>
          <p><code><a href="dep_gene2cat.html">dep_gene2cat</a></code> </p>
        </td>
        <td><p>GO terms assigned to proteins in DEP results</p></td>
      </tr><tr><td>
          <p><code><a href="dep_results.html">dep_results</a></code> </p>
        </td>
        <td><p>Results table from DEP package</p></td>
      </tr><tr><td>
          <p><code><a href="download_ccp_crap.html">download_ccp_crap()</a></code> </p>
        </td>
        <td><p>Download CCP cRAP FASTA file</p></td>
      </tr><tr><td>
          <p><code><a href="estimate_go_overrep.html">estimate_go_overrep()</a></code> </p>
        </td>
        <td><p>Estimate effect size of GO over-representation</p></td>
      </tr><tr><td>
          <p><code><a href="estimate_incorporation.html">estimate_incorporation()</a></code> </p>
        </td>
        <td><p>Estimate SILAC (2-plex) incorporation rate from PD output</p></td>
      </tr><tr><td>
          <p><code><a href="filter_TMT_PSMs.html">filter_TMT_PSMs()</a></code> <code><a href="filter_TMT_PSMs.html">filter_TMT_PSMs()</a></code> </p>
        </td>
        <td><p>Filter a PSM-level MSnSet to remove low quality PSMs</p></td>
      </tr><tr><td>
          <p><code><a href="get_ancestor_go.html">get_ancestor_go()</a></code> </p>
        </td>
        <td><p>Get all ancestor GO terms</p></td>
      </tr><tr><td>
          <p><code><a href="get_cat_palette.html">get_cat_palette()</a></code> </p>
        </td>
        <td><p>Generate a colour-blind friendly palette for categorical colour encoding</p></td>
      </tr><tr><td>
          <p><code><a href="get_ccp_crap.html">get_ccp_crap()</a></code> </p>
        </td>
        <td><p>Get the UniProt sequences in CCP cRAP</p></td>
      </tr><tr><td>
          <p><code><a href="get_enriched_go.html">get_enriched_go()</a></code> </p>
        </td>
        <td><p>GO term enrichment using goseq</p></td>
      </tr><tr><td>
          <p><code><a href="get_incorporation.html">get_incorporation()</a></code> </p>
        </td>
        <td><p>Determine SILAC (2-plex) incorporation rate given intensity values for Light and Heavy</p></td>
      </tr><tr><td>
          <p><code><a href="get_medians.html">get_medians()</a></code> </p>
        </td>
        <td><p>Extract the expression matrix column medians from an MSnSet</p></td>
      </tr><tr><td>
          <p><code><a href="get_notch_per_protein.html">get_notch_per_protein()</a></code> </p>
        </td>
        <td><p>Summarise the number of sub-notch PSM intensities per protein</p></td>
      </tr><tr><td>
          <p><code><a href="get_parsimony_pep2prot.html">get_parsimony_pep2prot()</a></code> </p>
        </td>
        <td><p>Identify the consensus peptide to master protein mapping across multiple runs</p></td>
      </tr><tr><td>
          <p><code><a href="get_psm_metrics.html">get_psm_metrics()</a></code> </p>
        </td>
        <td><p>Summarise percentage of tag intensities below a given threshold and missing</p></td>
      </tr><tr><td>
          <p><code><a href="get_ratio.html">get_ratio()</a></code> </p>
        </td>
        <td><p>Get ratio of two numbers</p></td>
      </tr><tr><td>
          <p><code><a href="get_sequence.html">get_sequence()</a></code> </p>
        </td>
        <td><p>Get the amino acid sequence around a PTM</p></td>
      </tr><tr><td>
          <p><code><a href="lfq_oops_rnase_pep.html">lfq_oops_rnase_pep</a></code> </p>
        </td>
        <td><p>PeptideGroups PD output for LFQ of RNase +/- OOPS experiment</p></td>
      </tr><tr><td>
          <p><code><a href="lfq_oops_rnase_psm.html">lfq_oops_rnase_psm</a></code> </p>
        </td>
        <td><p>PSM PD output for LFQ of RNase +/- OOPS experiment</p></td>
      </tr><tr><td>
          <p><code><a href="make_fasta.html">make_fasta()</a></code> </p>
        </td>
        <td><p>Make a FASTA using UniProt accessions</p></td>
      </tr><tr><td>
          <p><code><a href="parse_PTM_scores.html">parse_PTM_scores()</a></code> </p>
        </td>
        <td><p>Parse the PTM probabilities and add new columns with PTM information</p></td>
      </tr><tr><td>
          <p><code><a href="parse_features.html">parse_features()</a></code> </p>
        </td>
        <td><p>Parse Proteome Discoverer output</p></td>
      </tr><tr><td>
          <p><code><a href="pep_oops_1.html">pep_oops_1</a></code> </p>
        </td>
        <td><p>PeptideGroups PD output for SILAC labelled U-2 OS cells</p></td>
      </tr><tr><td>
          <p><code><a href="pep_oops_2.html">pep_oops_2</a></code> </p>
        </td>
        <td><p>PeptideGroups PD output for SILAC labelled U-2 OS cells</p></td>
      </tr><tr><td>
          <p><code><a href="pep_oops_3.html">pep_oops_3</a></code> </p>
        </td>
        <td><p>PeptideGroups PD output for SILAC labelled U-2 OS cells</p></td>
      </tr><tr><td>
          <p><code><a href="pep_oops_4.html">pep_oops_4</a></code> </p>
        </td>
        <td><p>PeptideGroups PD output for SILAC labelled U-2 OS cells</p></td>
      </tr><tr><td>
          <p><code><a href="pep_silac_p0.html">pep_silac_p0</a></code> </p>
        </td>
        <td><p>PeptideGroups PD output for SILAC labelled MOLM-13 cells</p></td>
      </tr><tr><td>
          <p><code><a href="pep_silac_p4.html">pep_silac_p4</a></code> </p>
        </td>
        <td><p>PeptideGroups PD output for SILAC labelled MOLM-13 cells</p></td>
      </tr><tr><td>
          <p><code><a href="plot_TMT_notch.html">plot_TMT_notch()</a></code> </p>
        </td>
        <td><p>Plot histograms for TMT tag intensities including notch annotation</p></td>
      </tr><tr><td>
          <p><code><a href="plot_below_notch_per_prot.html">plot_below_notch_per_prot()</a></code> </p>
        </td>
        <td><p>Plot the number of sub-notch PSM intensities per protein</p></td>
      </tr><tr><td>
          <p><code><a href="plot_fraction_below_notch_per_prot.html">plot_fraction_below_notch_per_prot()</a></code> </p>
        </td>
        <td><p>Plot the fraction of sub-notch PSM intensities per protein</p></td>
      </tr><tr><td>
          <p><code><a href="plot_go.html">plot_go()</a></code> </p>
        </td>
        <td><p>Plot selected GO terms</p></td>
      </tr><tr><td>
          <p><code><a href="plot_incorporation.html">plot_incorporation()</a></code> </p>
        </td>
        <td><p>Plot annotated histogram of incorporation values</p></td>
      </tr><tr><td>
          <p><code><a href="plot_missing_SN.html">plot_missing_SN()</a></code> </p>
        </td>
        <td><p>Plot the missing values vs signal:noise</p></td>
      </tr><tr><td>
          <p><code><a href="plot_missing_SN_per_sample.html">plot_missing_SN_per_sample()</a></code> </p>
        </td>
        <td><p>Plot the missing values vs signal:noise for each sample</p></td>
      </tr><tr><td>
          <p><code><a href="plot_quant.html">plot_quant()</a></code> </p>
        </td>
        <td><p>Plot distributions for feature intensities per sample.</p></td>
      </tr><tr><td>
          <p><code><a href="psm_oops_1.html">psm_oops_1</a></code> </p>
        </td>
        <td><p>PSM PD output for SILAC labelled U-2 OS cells</p></td>
      </tr><tr><td>
          <p><code><a href="psm_oops_2.html">psm_oops_2</a></code> </p>
        </td>
        <td><p>PSM PD output for SILAC labelled U-2 OS cells</p></td>
      </tr><tr><td>
          <p><code><a href="psm_oops_3.html">psm_oops_3</a></code> </p>
        </td>
        <td><p>PSM PD output for SILAC labelled U-2 OS cells</p></td>
      </tr><tr><td>
          <p><code><a href="psm_oops_4.html">psm_oops_4</a></code> </p>
        </td>
        <td><p>PSM PD output for SILAC labelled U-2 OS cells</p></td>
      </tr><tr><td>
          <p><code><a href="psm_silac_p4.html">psm_silac_p4</a></code> </p>
        </td>
        <td><p>PSMs PD output for SILAC labelled MOLM-13 cells</p></td>
      </tr><tr><td>
          <p><code><a href="psm_tmt_phospho.html">psm_tmt_phospho</a></code> </p>
        </td>
        <td><p>PSM-level PD output for phospho TMT10-plex data</p></td>
      </tr><tr><td>
          <p><code><a href="psm_tmt_total.html">psm_tmt_total</a></code> </p>
        </td>
        <td><p>PSM-level PD output for total proteome TMT10-plex data</p></td>
      </tr><tr><td>
          <p><code><a href="psm_to_peptide_style_modifications.html">psm_to_peptide_style_modifications()</a></code> </p>
        </td>
        <td><p>Transformes PSM-level Modifications annotation style to peptide-level style</p></td>
      </tr><tr><td>
          <p><code><a href="remove_dots.html">remove_dots()</a></code> </p>
        </td>
        <td><p>Remove duplicated full stops</p></td>
      </tr><tr><td>
          <p><code><a href="remove_redundant_go.html">remove_redundant_go()</a></code> </p>
        </td>
        <td><p>Remove redundant GO terms</p></td>
      </tr><tr><td>
          <p><code><a href="remove_silac_modifications.html">remove_silac_modifications()</a></code> </p>
        </td>
        <td><p>Remove SILAC labels from modifications</p></td>
      </tr><tr><td>
          <p><code><a href="remove_x.html">remove_x()</a></code> </p>
        </td>
        <td><p>Remove leading X</p></td>
      </tr><tr><td>
          <p><code><a href="restrict_features_per_protein.html">restrict_features_per_protein()</a></code> </p>
        </td>
        <td><p>Remove features which are assigned to a protein with too few supporting
features in total</p></td>
      </tr><tr><td>
          <p><code><a href="silac_psm_seq_int.html">silac_psm_seq_int()</a></code> </p>
        </td>
        <td><p>Extract peptide sequencing and interference information from SILAC
PSM-level output</p></td>
      </tr><tr><td>
          <p><code><a href="sub_crap.html">sub_crap()</a></code> </p>
        </td>
        <td><p>Insert cRAP numbers into a character vector</p></td>
      </tr><tr><td>
          <p><code><a href="theme_camprot.html">theme_camprot()</a></code> </p>
        </td>
        <td><p>ggplot theme for package</p></td>
      </tr><tr><td>
          <p><code><a href="theme_csd.html">theme_csd()</a></code> </p>
        </td>
        <td><p>Charlotte's ggplot2 theme</p></td>
      </tr><tr><td>
          <p><code><a href="update_average_sn.html">update_average_sn()</a></code> </p>
        </td>
        <td><p>Adds new feature describing the average reporter Signal/Noise ratio.</p></td>
      </tr></tbody></table></div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Charlotte Dawson, Tom Smith.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

