<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>TMT QC PSM-level quantification and summarisation to protein-level abundance • camprotR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="TMT QC PSM-level quantification and summarisation to protein-level abundance">
<meta property="og:description" content="camprotR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">camprotR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/TMT_PSM_QC_Summarisation.html">TMT QC PSM-level quantification and summarisation to protein-level abundance</a>
    </li>
    <li>
      <a href="../articles/crap.html">cRAP databases</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="TMT_PSM_QC_Summarisation_files/header-attrs-2.10/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>TMT QC PSM-level quantification and summarisation to protein-level abundance</h1>
                        <h4 class="author">Tom Smith</h4>
            
            <h4 class="date">2021-08-17</h4>
      
      
      <div class="hidden name"><code>TMT_PSM_QC_Summarisation.Rmd</code></div>

    </div>

    
    
<p>Quantitative proteomics using isobaric tagging such as Tandem Mass Tags (TMT) has a considerable benefit over Label-Free Quantification (LFQ) in that up to 16 samples can be quantified for each Peptide Spectrum Match (PSM). This greatly reduces the extent of missing values that may be present between different MS runs due to the limited number of ions that can be fragmented in each run and the associated issue of peptides being identified in only a subset of runs <span class="citation">(O’Connell et al. 2018)</span>. As such, it standardises the features quantified in each sample, simplifying the comparison between samples and increasing quantification accuracy of summarised features such as proteins.</p>
<p>However, TMT does suffer from ratio compression, which should be avoiding by performing quantification with SPS MS3 <span class="citation">(McAlister et al. 2014)</span>. Since quantification is performed in MS3, the TMT ions being quantified are therefore relatively low intensity and as such, careful QC of the PSM-level quantification is recommended prior to summarisation to protein-level abundances.</p>
<p>Here, we will assess the following:</p>
<ul>
<li><p>The overall distribution of tag intensities, including the so-called ‘notch’ region <span class="citation">(Hughes et al. 2017)</span></p></li>
<li><p>How signal:noise relates to missing values</p></li>
</ul>
<p>We will then:</p>
<ul>
<li><p>Filter the PSMs based on signal:noise and co-isolation/interference</p></li>
<li><p>Summarise to protein-level abundances</p></li>
</ul>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">camprotR</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://lgatto.github.io/MSnbase">MSnbase</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org">tidyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span></code></pre></div>
<p>We start by reading in a file containing PSM-level output from Proteome Discoverer (PD). Here we are using a truncated file containing the first 5,000 PSMs only.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">psm_data</span> <span class="op">&lt;-</span> <span class="va">psm_tmt_total</span></code></pre></div>
<p>The first step is to remove contaminant proteins. These were defined using the cRAP database (REF). Below, we parse the cRAP fasta to extract the IDs for the cRAP proteins, in both ‘cRAP’ format and Uniprot IDs for these proteins.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">crap_fasta_inf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span>
  <span class="st">"extdata"</span>, <span class="st">"cRAP_FullIdentifiers.fasta.gz"</span>, 
  package <span class="op">=</span> <span class="st">"camprotR"</span>
<span class="op">)</span>

<span class="co"># Load the cRAP FASTA used for the PD search</span>
<span class="va">crap.fasta</span> <span class="op">&lt;-</span> <span class="fu">Biostrings</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html">fasta.index</a></span><span class="op">(</span><span class="va">crap_fasta_inf</span>, seqtype <span class="op">=</span> <span class="st">"AA"</span><span class="op">)</span>

<span class="co"># Define a base R version of stringr::str_extract_all()</span>
<span class="co"># base R str_extract</span>
<span class="va">str_extract_all</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">pattern</span>, <span class="va">string</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gregexpr</a></span><span class="op">(</span><span class="va">pattern</span>, <span class="va">string</span>, perl <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">%&gt;%</span> 
    <span class="fu"><a href="https://rdrr.io/r/base/regmatches.html">regmatches</a></span><span class="op">(</span><span class="va">string</span>, <span class="va">.</span><span class="op">)</span> <span class="op">%&gt;%</span> 
    <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="op">)</span>
<span class="op">}</span>

<span class="co"># Extract the non cRAP UniProt accessions associated with each cRAP protein</span>
<span class="va">crap.accessions</span> <span class="op">&lt;-</span> <span class="va">crap.fasta</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span><span class="op">(</span><span class="va">desc</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu">str_extract_all</span><span class="op">(</span><span class="st">"(?&lt;=\\|).*?(?=\\|)"</span>, <span class="va">.</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>We can then supply these cRAP protein IDs to <code>parse_features</code> which will remove features which may originate from contaminants, as well as features which don’t have a unique master protein. See <code><a href="../reference/parse_features.html">?parse_features</a></code> for further details, including the removal of ‘associated cRAP.’</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">psm_data_flt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/parse_features.html">parse_features</a></span><span class="op">(</span>
  <span class="va">psm_data</span>, 
  crap_proteins <span class="op">=</span> <span class="va">crap.accessions</span>, 
  TMT <span class="op">=</span> <span class="cn">TRUE</span>, 
  level <span class="op">=</span> <span class="st">'PSM'</span>
<span class="op">)</span>
<span class="co">#&gt; Parsing features...</span>
<span class="co">#&gt; 5000 features found from 2364 master proteins =&gt; Input</span>
<span class="co">#&gt; 230 cRAP proteins supplied</span>
<span class="co">#&gt; 60 proteins identified as 'cRAP associated'</span>
<span class="co">#&gt; 4933 features found from 2346 master proteins =&gt; cRAP features removed</span>
<span class="co">#&gt; 4924 features found from 2340 master proteins =&gt; associated cRAP features removed</span>
<span class="co">#&gt; 4922 features found from 2339 master proteins =&gt; features without a master protein removed</span>
<span class="co">#&gt; 4748 features found from 2242 master proteins =&gt; features with non-unique master proteins removed</span></code></pre></div>
<p>We now store the filtered PSM data in an MSnSet, the standard data object for proteomics in R.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Abundance columns for TMT PD-output start with Abundance </span>
<span class="va">abundance_cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">psm_data_flt</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">'Abundance.'</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">psm_data_flt</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>

<span class="va">psm.e</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">psm_data_flt</span><span class="op">[</span>, <span class="va">abundance_cols</span><span class="op">]</span><span class="op">)</span>
<span class="va">psm.f</span> <span class="op">&lt;-</span> <span class="va">psm_data_flt</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/pkg/generics/man/setops.html">setdiff</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">psm_data_flt</span><span class="op">)</span>, <span class="va">abundance_cols</span><span class="op">)</span><span class="op">]</span>

<span class="co"># update the column names to remove the 'Abundance.` prefix</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">psm.e</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">'Abundance.'</span>, <span class="st">''</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">psm.e</span><span class="op">)</span><span class="op">)</span>

<span class="co"># we don't have 'phenotype' data to add so we just define the </span>
<span class="co"># 'expression' data and 'feature' data</span>
<span class="va">psm</span> <span class="op">&lt;-</span> <span class="fu">MSnbase</span><span class="fu">::</span><span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/MSnSet-class.html">MSnSet</a></span><span class="op">(</span>exprs <span class="op">=</span> <span class="va">psm.e</span>, fData <span class="op">=</span> <span class="va">psm.f</span><span class="op">)</span></code></pre></div>
<p>The distribution of TMT intensities obtained by Orbitrap has been observed to have a ‘notch’ where few values are observed <span class="citation">(Hughes et al. 2017)</span>. This presence of the notch is dependent on the MS settings, most notably automatic gain control (AGC) target and maximum injection time, and the abundance of the peptides injected. We can inspect the notch using the <code>plot_TMT_notch</code> function, which annotates the notch region and the proportion of values below the upper notch boundary.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_TMT_notch.html">plot_TMT_notch</a></span><span class="op">(</span><span class="va">psm</span><span class="op">)</span>
<span class="co">#&gt; Warning: Removed 587 rows containing non-finite values (stat_bin).</span></code></pre></div>
<div class="figure">
<img src="TMT_PSM_QC_Summarisation_files/figure-html/unnamed-chunk-6-1.png" alt="TMT intensities, with notch region denoted by vertical lines" width="480"><p class="caption">
TMT intensities, with notch region denoted by vertical lines
</p>
</div>
<p>Sub-notch values are systematic under-estimates (currently unpublished findings). The impact of sub-notch values on summarised protein-level abundances will depend on the number of sub-notch values and, more importantly, the proportion of PSMs below the notch for a given protein. We can obtain the per-protein notch metrics using <code>get_notch_per_protein</code> and plot the number and fraction with <code>plot_below_notch_per_prot</code> and <code>plot_fraction_below_notch_per_prot</code> respectively.</p>
<p>In the case, we see that most proteins have zero PSM intensities below the notch. The vast majority of protein have &lt;10% PSMs at/below the notch, indicating the notch is unlikely to be problematic with this dataset.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">notch_per_protein</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_notch_per_protein.html">get_notch_per_protein</a></span><span class="op">(</span><span class="va">psm</span><span class="op">)</span>

<span class="fu"><a href="../reference/plot_below_notch_per_prot.html">plot_below_notch_per_prot</a></span><span class="op">(</span><span class="va">notch_per_protein</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="TMT_PSM_QC_Summarisation_files/figure-html/unnamed-chunk-7-1.png" alt="Count of TMT intensities below the notch per protein" width="480"><p class="caption">
Count of TMT intensities below the notch per protein
</p>
</div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_fraction_below_notch_per_prot.html">plot_fraction_below_notch_per_prot</a></span><span class="op">(</span><span class="va">notch_per_protein</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="TMT_PSM_QC_Summarisation_files/figure-html/unnamed-chunk-8-1.png" alt="Fraction of TMT intensities below the notch per protein" width="480"><p class="caption">
Fraction of TMT intensities below the notch per protein
</p>
</div>
<p>We will still want to remove low Signal:Noise (S:N) PSMs, since the quantification values will be less accurate and there will be more missing values and sub-notch values. We can inspect the relationship between S:N and missing values using the <code>plot_missing_SN</code> function.</p>
<p>Note that where the signal:noise &gt; 10, there are far fewer missing values.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_missing_SN.html">plot_missing_SN</a></span><span class="op">(</span><span class="va">psm</span>, bins <span class="op">=</span> <span class="fl">27</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="TMT_PSM_QC_Summarisation_files/figure-html/unnamed-chunk-9-1.png" alt="Missing values per PSM, in relation to the signal:noise ratio" width="672"><p class="caption">
Missing values per PSM, in relation to the signal:noise ratio
</p>
</div>
<p>We can also look into this relationship at the tag level using <code>plot_missing_SN_per_sample</code>. In this case, there is no tag which appears to have a high proportion of missing values when signal:noise &gt; 10.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_missing_SN_per_sample.html">plot_missing_SN_per_sample</a></span><span class="op">(</span><span class="va">psm</span>, bins <span class="op">=</span> <span class="fl">27</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="TMT_PSM_QC_Summarisation_files/figure-html/unnamed-chunk-10-1.png" alt="Missing values per tag, in relation to the signal:noise ratio" width="672"><p class="caption">
Missing values per tag, in relation to the signal:noise ratio
</p>
</div>
<p>Based on the above, we will filter the PSMs to only retain those with S:N &gt; 10 using <code>filter_TMT_PSMs</code>. Using the same function, we will also remove PSMs with interference/co-isolation &gt;50%.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">psm_filt_sn_int</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_TMT_PSMs.html">filter_TMT_PSMs</a></span><span class="op">(</span><span class="va">psm</span>, inter_thresh <span class="op">=</span> <span class="fl">50</span>, sn_thresh <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>
<span class="co">#&gt; Filtering PSMs...</span>
<span class="co">#&gt; 4747 features found from 2242 master proteins =&gt; No quant filtering</span>
<span class="co">#&gt; 4628 features found from 2206 master proteins =&gt; Co-isolation filtering</span>
<span class="co">#&gt; 4453 features found from 2174 master proteins =&gt; S:N ratio filtering</span></code></pre></div>
<p>We can then re-plot to see how this affects the notch.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_TMT_notch.html">plot_TMT_notch</a></span><span class="op">(</span><span class="va">psm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">'Before filtering'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/theme_camprot.html">theme_camprot</a></span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>
<span class="co">#&gt; Warning: Removed 587 rows containing non-finite values (stat_bin).</span>

<span class="fu"><a href="../reference/plot_TMT_notch.html">plot_TMT_notch</a></span><span class="op">(</span><span class="va">psm_filt_sn_int</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">'After filtering'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/theme_camprot.html">theme_camprot</a></span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>
<span class="co">#&gt; Warning: Removed 224 rows containing non-finite values (stat_bin).</span></code></pre></div>
<div class="figure">
<img src="TMT_PSM_QC_Summarisation_files/figure-html/unnamed-chunk-12-1.png" alt="Notch plots after filtering" width="326.4"><img src="TMT_PSM_QC_Summarisation_files/figure-html/unnamed-chunk-12-2.png" alt="Notch plots after filtering" width="326.4"><p class="caption">
Notch plots after filtering
</p>
</div>
<p>Now that we have inspected the PSM-level quantification and filtered the PSMs, we can summarise the PSMs to protein-level abundances.</p>
<p>For PSM to protein summarisation, we will use <code><a href="https://lgatto.github.io/MSnbase/reference/combineFeatures.html">MSnbase::combineFeatures(method = 'robust')</a></code>. This approach can handle missing values but it only makes sense to use PSMs quantified across enough proteins and to retain proteins with enough PSMs.</p>
<p>Here, we will use PSMs with at most 5/10 <code>NA</code>s and proteins with at least three finite quantification values in a given sample. Note that this means a given protein may still have a mixture of finite and <code>NA</code> values. For example, if a protein has 3 PSMs and the PSMs have missing values such that some samples are still quantified in all 3 PSMs, then the protein will be quantified in these samples, but <code>NA</code> where missing values are present. Thus, each protein-level quantification will be derived from at least 3 PSM intensities, where each PSM was quantified in at least 5 samples.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">psm_filt_sn_int_missing</span> <span class="op">&lt;-</span> <span class="va">psm_filt_sn_int</span> <span class="op">%&gt;%</span> 
  <span class="fu">MSnbase</span><span class="fu">::</span><span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/MSnSet-class.html">filterNA</a></span><span class="op">(</span>pNA <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="co"># remove features observed in fewer than </span>
  <span class="fu"><a href="../reference/restrict_features_per_protein.html">restrict_features_per_protein</a></span><span class="op">(</span>min_features <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>
<span class="co">#&gt; `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre></div>
<div class="figure">
<img src="TMT_PSM_QC_Summarisation_files/figure-html/unnamed-chunk-13-1.png" alt="Tally of PSMs per protein" width="480"><p class="caption">
Tally of PSMs per protein
</p>
</div>
<p>Below, we perform the summarisation. Note that robust summarisation requires log-transformed intensities. If desired, one can exponeniate the protein-level intensities post-summarisation, as shown below.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">prot_group</span> <span class="op">&lt;-</span> <span class="fu">MSnbase</span><span class="fu">::</span><span class="fu">fData</span><span class="op">(</span><span class="va">psm_filt_sn_int_missing</span><span class="op">)</span><span class="op">$</span><span class="va">Master.Protein.Accessions</span>

<span class="va">protein</span> <span class="op">&lt;-</span> <span class="va">psm_filt_sn_int_missing</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log</a></span><span class="op">(</span>base <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">MSnbase</span><span class="fu">::</span><span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/combineFeatures.html">combineFeatures</a></span><span class="op">(</span>groupBy <span class="op">=</span> <span class="va">prot_group</span>, method <span class="op">=</span> <span class="st">'robust'</span>, maxit <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span>
<span class="co">#&gt; Your data contains missing values. Please read the relevant section in</span>
<span class="co">#&gt; the combineFeatures manual page for details on the effects of missing</span>
<span class="co">#&gt; values on data aggregation.</span>

<span class="fu">Biobase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span><span class="op">(</span><span class="va">protein</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fl">2</span><span class="op">^</span><span class="fu">Biobase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span><span class="op">(</span><span class="va">protein</span><span class="op">)</span></code></pre></div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-http://zotero-org/users/5634351/items/DLWEN74B" class="csl-entry">
Hughes, Christopher S., Chenchen Zhu, Victor Spicer, Oleg V. Krokhin, and Gregg B. Morin. 2017. <span>“Evaluating the Characteristics of Reporter Ion Signal Acquired in the Orbitrap Analyzer for Isobaric Mass Tag Proteome Quantification Experiments.”</span> <em>Journal of Proteome Research</em> 16 (5): 1831–38. <a href="https://doi.org/10.1021/acs.jproteome.7b00092">https://doi.org/10.1021/acs.jproteome.7b00092</a>.
</div>
<div id="ref-http://zotero-org/users/5634351/items/ZK69WYZ2" class="csl-entry">
McAlister, Graeme C., David P. Nusinow, Mark P. Jedrychowski, Martin Wühr, Edward L. Huttlin, Brian K. Erickson, Ramin Rad, Wilhelm Haas, and Steven P. Gygi. 2014. <span>“MultiNotch Ms3 Enables Accurate, Sensitive, and Multiplexed Detection of Differential Expression Across Cancer Cell Line Proteomes.”</span> <em>Analytical Chemistry</em> 86 (14): 7150–58. <a href="https://doi.org/10.1021/ac502040v">https://doi.org/10.1021/ac502040v</a>.
</div>
<div id="ref-http://zotero-org/users/5634351/items/ME322P2A" class="csl-entry">
O’Connell, Jeremy D., Joao A. Paulo, Jonathon J. O’Brien, and Steven P. Gygi. 2018. <span>“Proteome-Wide Evaluation of Two Common Protein Quantification Methods.”</span> <em>Journal of Proteome Research</em> 17 (5): 1934–42. <a href="https://doi.org/10.1021/acs.jproteome.8b00016">https://doi.org/10.1021/acs.jproteome.8b00016</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Charlotte Dawson, Tom Smith.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
