<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>cRAP databases • camprotR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="cRAP databases">
<meta property="og:description" content="camprotR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">camprotR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/LFQ_parse_qc_protein_inference.html">LFQ QC and summarisation to protein-level abundance</a>
    </li>
    <li>
      <a href="../articles/SILAC.html">Processing and QC of SILAC data</a>
    </li>
    <li>
      <a href="../articles/TMT_PSM_QC_Summarisation.html">TMT QC PSM-level quantification and summarisation to protein-level abundance</a>
    </li>
    <li>
      <a href="../articles/crap.html">cRAP databases</a>
    </li>
    <li>
      <a href="../articles/msnset.html">Working with MSnSets</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>cRAP databases</h1>
                        <h4 data-toc-skip class="author">Charlotte
Dawson</h4>
            
            <h4 data-toc-skip class="date">2023-01-17</h4>
      
      
      <div class="hidden name"><code>crap.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biostrings" class="external-link">Biostrings</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">camprotR</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://httr.r-lib.org/" class="external-link">httr</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Peptide-spectrum matching requires 2 main inputs: a collection of MS2
spectra and a FASTA database of protein/peptide sequences. Search
algorithms are then used to match the supplied MS2 spectra to
theoretical spectra generated from an in-silico digest of the supplied
FASTA database (typically we use the SwissProt protein database for the
organism of interest). The selection of the FASTA database should be
done carefully as having a very large database can reduce the number of
PSMs obtained <span class="citation">(<strong>Jeong2012?</strong>)</span>. Supposedly, using
an unecessarily large database can also affect <a href="https://pwilmart.github.io/blog/2020/09/19/shotgun-quantification-part2" class="external-link">protein
quantification</a>.</p>
<p>There are certain proteinaceous contaminants that are commonly,
accidentally introduced during sample preparation for proteomics. In
order to avoid misidentification of MS2 spectra that have come from
these contaminants we use an additional (much smaller) FASTA database
containing the sequences of these common contaminants. This is generally
called a contaminants database, or is colloquially known as a cRAP
database in our lab.</p>
<p><strong>It is very important that you know what proteins are in your
contaminants database as PSMs/peptides from these proteins are
deliberately filtered out in many of the functions in
<code>camprotR</code></strong>. See this <a href="http://proteomicsnews.blogspot.com/2020/12/is-one-of-your-proteins-of-interest-in.html" class="external-link">blog
post</a> some experimental situations where you may be interested in
proteins that are usually deemed contaminants.</p>
</div>
<div class="section level2">
<h2 id="the-3-crap-databases">The 3 cRAP databases<a class="anchor" aria-label="anchor" href="#the-3-crap-databases"></a>
</h2>
<p>There are two widely used FASTA databases containing common
contaminants:</p>
<ol style="list-style-type: decimal">
<li>The Global Proteome Machine (GPM) common Repository of Adventitious
Proteins (cRAP)</li>
<li>The <code>contaminants.fasta</code> file which is distributed in
every single installation of MaxQuant.</li>
</ol>
<p>In addition, in the CCP we use our own cRAP database which is similar
but not identical to the GPM cRAP. <strong>I highly recommend you use
the CCP cRAP database, as the other two are quite out of date</strong>.
The <code>download_crap()</code> function in <code>camprotR</code>
provides an easy and quick way to download the CCP cRAP database (see
the section below).</p>
<div class="section level3">
<h3 id="common-repository-of-adventitious-proteins-crap">common Repository of Adventitious Proteins (cRAP)<a class="anchor" aria-label="anchor" href="#common-repository-of-adventitious-proteins-crap"></a>
</h3>
<p>This list of proteins was collated by the Global Proteome Machine
(GPM) organisation. To quote the <a href="https://www.thegpm.org/crap/" class="external-link">cRAP website</a>,</p>
<blockquote>
<p>cRAP is an attempt to create a list of proteins commonly found in
proteomics experiments either by accident or through unavoidable
contamination of protein samples.</p>
</blockquote>
<p>cRAP includes 5 main categories of proteins:</p>
<ol style="list-style-type: decimal">
<li>Laboratory proteins used somewhere in the sample preparation
process.</li>
<li>Proteins added from dust or accidental contact with the sample</li>
<li>Proteins used as molecular weight markers</li>
<li>Proteins in the Sigma Universal Protein Standard (UPS)</li>
<li>Common viral contaminants</li>
</ol>
<p>According to the cRAP website, it has not been updated since 2012.
Therefore, I would not recommend using this FASTA for your peptide
searches.</p>
<p>Nevertheless, we can download this FASTA database through R. The
following chunk is not run by default as the URL is not always
stable.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># download cRAP</span></span>
<span><span class="va">crap_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span>fileext <span class="op">=</span> <span class="st">".fasta"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"ftp://ftp.thegpm.org/fasta/cRAP/crap.fasta"</span>,</span>
<span>              destfile <span class="op">=</span> <span class="va">crap_path</span><span class="op">)</span></span></code></pre></div>
<p>The GPM cRAP database is also installed with <code>camprotR</code>.
You can access it with the following command.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">crap_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"cRAP_GPM.fasta.gz"</span>, package <span class="op">=</span> <span class="st">"camprotR"</span><span class="op">)</span></span></code></pre></div>
<p>Then we can take a look at it using the useful
<code>Biostrings</code> package. With <code>Biostrings</code> we can
load FASTA files into R and easily compare and manipulate them.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load cRAP FASTA</span></span>
<span><span class="va">crap</span> <span class="op">&lt;-</span> <span class="fu">Biostrings</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html" class="external-link">readAAStringSet</a></span><span class="op">(</span><span class="va">crap_path</span><span class="op">)</span></span>
<span><span class="va">crap</span></span>
<span><span class="co">#&gt; AAStringSet object of length 116:</span></span>
<span><span class="co">#&gt;       width seq                                             names               </span></span>
<span><span class="co">#&gt;   [1]   607 MKWVTFISLLLLFSSAYSRGVF...DKEACFAVEGPKLVVSTQTALA sp|ALBU_BOVIN|</span></span>
<span><span class="co">#&gt;   [2]   511 MKLFWLLFTIGFCWAQYSSNTQ...AHFSISNSAEDPFIAIHAESKL sp|AMYS_HUMAN|</span></span>
<span><span class="co">#&gt;   [3]   214 MKLLILTCLVAVALARPKHPIK...SFSDIPNPIGSENSEKTTMPLW sp|CAS1_BOVIN|</span></span>
<span><span class="co">#&gt;   [4]   222 MKFFIFTCLLAVALAKNTMEHV...HQKAMKPWIQPKTKVIPYVRYL sp|CAS2_BOVIN|</span></span>
<span><span class="co">#&gt;   [5]   224 MKVLILACLVALALARELEELN...QAFLLYQEPVLGPVRGPFPIIV sp|CASB_BOVIN|</span></span>
<span><span class="co">#&gt;   ...   ... ...</span></span>
<span><span class="co">#&gt; [112]   238 MSKGEELFTGVVPILVELDGDV...HMVLLEFVTAAGITHGMDELYK sp|GFP_AEQVI|</span></span>
<span><span class="co">#&gt; [113]   204 MAEEVEEERLKYLDFVRAAGVY...VSSYLPLLPTEKITKVFGDEAS sp|SRPP_HEVBR|</span></span>
<span><span class="co">#&gt; [114]   138 MAEDEDNQQGQGEGLKYLGFVQ...RSLASSLPGQTKILAKVFYGEN sp|REF_HEVBR|</span></span>
<span><span class="co">#&gt; [115]   348 MFSSVMVALVSLAVAVSANPGL...PDKAVMNADNHEYFSENNPAQS sp|PLMP_GRIFR|</span></span>
<span><span class="co">#&gt; [116]   271 MSHIQRETSCSRPRLNSNLDAD...KYGIDNPDMNKLQFHLMLDEFF KKA1_ECOLX</span></span></code></pre></div>
<p>We can see that there are 116 sequences in cRAP.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">crap</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "sp|ALBU_BOVIN|" "sp|AMYS_HUMAN|" "sp|CAS1_BOVIN|" "sp|CAS2_BOVIN|"</span></span>
<span><span class="co">#&gt; [5] "sp|CASB_BOVIN|" "sp|CASK_BOVIN|"</span></span></code></pre></div>
<p>We can see that the headers for each sequence in the FASTA file have
very little information in them. Another reason not to use this cRAP
database as your contaminants database.</p>
</div>
<div class="section level3">
<h3 id="cambridge-centre-for-proteomics-crap-ccp-crap">Cambridge Centre for Proteomics cRAP (CCP cRAP)<a class="anchor" aria-label="anchor" href="#cambridge-centre-for-proteomics-crap-ccp-crap"></a>
</h3>
<p>The cRAP database we use in the CCP is largely based off of the GPM
cRAP database, with a few extra sequences added on the end. It also has
much more informative headers and the sequences for the commercial
proteases Endoproteinase GluC (NEB, P8100S) and recombinant Lys-C
(Promega, V167A) have been added.</p>
<p>You can download the latest version of the CCP cRAP using the
<code><a href="../reference/download_ccp_crap.html">download_ccp_crap()</a></code> function. It is
<strong>important</strong> to take a note of what date you download the
CCP cRAP database and what the current UniProt release is, as the
sequences and accessions can change (slightly) over time. Generally I
like to include the UniProt release in the file name
e.g. <code>2021-01_CCP_cRAP.fasta</code>, but hfor this example we’ll
just use a temporary file.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ccp_tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span>fileext <span class="op">=</span> <span class="st">".fasta"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/download_ccp_crap.html">download_ccp_crap</a></span><span class="op">(</span><span class="va">ccp_tmp</span>, is_crap <span class="op">=</span> <span class="cn">TRUE</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Downloading from UniProtKB release: 2022_05</span></span></code></pre></div>
<p>We can load this FASTA file into R and take a look.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ccp_crap</span> <span class="op">&lt;-</span> <span class="fu">Biostrings</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html" class="external-link">readAAStringSet</a></span><span class="op">(</span><span class="va">ccp_tmp</span><span class="op">)</span></span>
<span><span class="va">ccp_crap</span></span>
<span><span class="co">#&gt; AAStringSet object of length 125:</span></span>
<span><span class="co">#&gt;       width seq                                             names               </span></span>
<span><span class="co">#&gt;   [1]   348 MSIPETQKGVIFYESHGKLEYK...PEIYEKMEKGQIVGRYVVDTSK sp|cRAP001|P00330...</span></span>
<span><span class="co">#&gt;   [2]   609 MKWVTFISLLFLFSSAYSRGVF...KETCFAEEGKKLVAASQAALGL sp|cRAP002|P02768...</span></span>
<span><span class="co">#&gt;   [3]   364 MPHSHPALTPEQKKELSDIAHR...YTPSGQAGAAASESLFISNHAY sp|cRAP003|P00883...</span></span>
<span><span class="co">#&gt;   [4]   511 MKLFWLLFTIGFCWAQYSSNTQ...AHFSISNSAEDPFIAIHAESKL sp|cRAP004|P0DUB6...</span></span>
<span><span class="co">#&gt;   [5]   511 MKLFWLLFTIGFCWAQYSSNTQ...AHFSISNSAEDPFIAIHAESKL sp|cRAP005|P0DTE7...</span></span>
<span><span class="co">#&gt;   ...   ... ...</span></span>
<span><span class="co">#&gt; [121]   387 MNVLLLLVLCTLAMGCGATSPP...KEQRSAECPGPAQKGYPFILPS sp|cRAP121|Q58D62...</span></span>
<span><span class="co">#&gt; [122]   266 MLRLLVFTSLVLYGHSTQDFPE...KPTVFTRVSAYISWINNAIASN sp|cRAP122|Q28153...</span></span>
<span><span class="co">#&gt; [123]   269 MIRALLLSTLVAGALSCGVPTY...KPSVFTRVSNYNDWISSVIENN sp|cRAP123|Q29461...</span></span>
<span><span class="co">#&gt; [124]   274 VILPNNDRHQITDTTNGHYAPV...NPDNGDNNNSDNPDAAHHHHHH sp|cRAP126|000000...</span></span>
<span><span class="co">#&gt; [125]   261 AGYRDGFGASGSCEVDAVCATQ...GVYSQISRYFAPHQHQHQHQHQ sp|cRAP127|000000...</span></span></code></pre></div>
<p>We can see there are 125 sequences. Now lets have a look at some of
the sequence headers.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ccp_crap</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "sp|cRAP001|P00330|ADH1_YEAST Alcohol dehydrogenase 1 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=ADH1 PE=1 SV=5"</span></span>
<span><span class="co">#&gt; [2] "sp|cRAP002|P02768|ALBU_HUMAN Albumin OS=Homo sapiens OX=9606 GN=ALB PE=1 SV=2"                                                            </span></span>
<span><span class="co">#&gt; [3] "sp|cRAP003|P00883|ALDOA_RABIT Fructose-bisphosphate aldolase A OS=Oryctolagus cuniculus OX=9986 GN=ALDOA PE=1 SV=2"                       </span></span>
<span><span class="co">#&gt; [4] "sp|cRAP004|P0DUB6|AMY1A_HUMAN Alpha-amylase 1A OS=Homo sapiens OX=9606 GN=AMY1A PE=1 SV=1"                                                </span></span>
<span><span class="co">#&gt; [5] "sp|cRAP005|P0DTE7|AMY1B_HUMAN Alpha-amylase 1B OS=Homo sapiens OX=9606 GN=AMY1B PE=1 SV=1"                                                </span></span>
<span><span class="co">#&gt; [6] "sp|cRAP006|P0DTE8|AMY1C_HUMAN Alpha-amylase 1C OS=Homo sapiens OX=9606 GN=AMY1C PE=1 SV=1"</span></span></code></pre></div>
<p>These sequence headers are actually complete, and follow the standard
structure of UniProt FASTA headers. <strong>Importantly</strong> they
contain both the UniProt accession e.g. P00330 <strong>and</strong> the
sequence version e.g. SV=5. Each time the amino acid sequence of a
UniProt entry is modified, the sequence version is incremented by 1.
With the UniProt accession and the SV number, one can know exactly which
sequence is being referred to in the FASTA file.</p>
<p>This isn’t exactly important for everyday use, but may prove
important when reanalysing a proteomics experiment after a long time has
passed.</p>
</div>
<div class="section level3">
<h3 id="maxquant-contaminants-fasta">MaxQuant contaminants.fasta<a class="anchor" aria-label="anchor" href="#maxquant-contaminants-fasta"></a>
</h3>
<p>Included in every installation of MaxQuant there is a file called
<code>contaminants.fasta</code>. This is the contaminants database that
MaxQuant will use be default for peptide searches, however it is
possible to provide your own contaminants FASTA in MaxQuant if you
want.</p>
<p>This is conjecture, but I’m fairly sure this FASTA database hasn’t
changed since the MaxQuant paper was released in 2008, making it
possibly even older than the GPM cRAP database. Therefore, I also would
not recommend using this FASTA for your peptide searches.</p>
<p>Nevertheless, we can download this FASTA database through R and
explore it like the GPM cRAP FASTA above. The following chunk is not run
by default as the URL is not always stable.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># download cRAP</span></span>
<span><span class="va">mq_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span>fileext <span class="op">=</span> <span class="st">".fasta"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"http://lotus1.gwdg.de/mpg/mmbc/maxquant_input.nsf/7994124a4298328fc125748d0048fee2/$FILE/contaminants.fasta"</span>,</span>
<span>              destfile <span class="op">=</span> <span class="va">mq_path</span><span class="op">)</span></span></code></pre></div>
<p>This FASTA is also installed with <code>camprotR</code> and you can
access it in a similar way to the GPM cRAP FASTA.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get file path</span></span>
<span><span class="va">mq_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"contaminants.fasta.gz"</span>, package <span class="op">=</span> <span class="st">"camprotR"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># load cRAP FASTA</span></span>
<span><span class="va">mq</span> <span class="op">&lt;-</span> <span class="fu">Biostrings</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html" class="external-link">readAAStringSet</a></span><span class="op">(</span><span class="va">mq_path</span><span class="op">)</span></span>
<span><span class="va">mq</span></span>
<span><span class="co">#&gt; AAStringSet object of length 245:</span></span>
<span><span class="co">#&gt;       width seq                                             names               </span></span>
<span><span class="co">#&gt;   [1]   231 FPTDDDDKIVGGYTCAANSIPY...KPGVYTKVCNYVNWIQQTIAAN P00761 SWISS-PROT...</span></span>
<span><span class="co">#&gt;   [2]   540 MSRQFTYKSGAAAKGGFSGCSA...SEFRDSQGKTLALSSPTKKTMR Q32MB2 TREMBL:Q32...</span></span>
<span><span class="co">#&gt;   [3]   594 MTSVGVFSDMLNGCGKDGLVPR...GGSVSGSSSSKIISTTTLNKRR P19013 SWISS-PROT...</span></span>
<span><span class="co">#&gt;   [4]   521 MSLSPCRAQRGFSARSACSARS...GSSCHTILKKTVESSLKTSITY Q7RTT2 TREMBL:Q7R...</span></span>
<span><span class="co">#&gt;   [5]   653 MKRICGSLLLLGLSISAALAAP...IIQTFTKDLSSEAAQRAPGSCG P15636 SWISS-PROT...</span></span>
<span><span class="co">#&gt;   ...   ... ...</span></span>
<span><span class="co">#&gt; [241]   199 MKPVSGRKSPVLYLLGILTVLL...LSLEIEKLEQEKRKKLQYQYAS ENSEMBL:ENSBTAP00...</span></span>
<span><span class="co">#&gt; [242]   452 MWAVLSLPLACLLAQAWLVPGS...IIYEETSRTVLFLGRVVDPTLL ENSEMBL:ENSBTAP00...</span></span>
<span><span class="co">#&gt; [243]   591 MSFGNWKPTVVVQGILWILYGL...RRASSRSKRRPPPTILLSTDLQ ENSEMBL:ENSBTAP00...</span></span>
<span><span class="co">#&gt; [244]  1477 MGKNKLLYPSLTLLLLLLLPTD...YETDEFAVAEYSAPCSKDIGNA ENSEMBL:ENSBTAP00...</span></span>
<span><span class="co">#&gt; [245]   604 MKQLKLTGFVIFFFFLTESLTL...AQEPEACFKEESPKLAAKSQAA REFSEQ:XP_585019 ...</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">mq</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "P00761 SWISS-PROT:P00761|TRYP_PIG Trypsin - Sus scrofa (Pig)."                           </span></span>
<span><span class="co">#&gt; [2] "Q32MB2 TREMBL:Q32MB2;Q86Y46 Tax_Id=9606 Gene_Symbol=KRT73 Keratin-73"                    </span></span>
<span><span class="co">#&gt; [3] "P19013 SWISS-PROT:P19013 Tax_Id=9606 Gene_Symbol=KRT4 keratin 4"                         </span></span>
<span><span class="co">#&gt; [4] "Q7RTT2 TREMBL:Q7RTT2 Tax_Id=9606 Gene_Symbol=KRT78 Keratin-78"                           </span></span>
<span><span class="co">#&gt; [5] "P15636 SWISS-PROT:P15636 Protease I precursor Lysyl endopeptidase Achromobacter lyticus."</span></span>
<span><span class="co">#&gt; [6] "P09870 SWISS-PROT:P09870 Arg-C (Clostripain) - Clostridium histolyticum."</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="make-your-own-crap-database">Make your own cRAP database<a class="anchor" aria-label="anchor" href="#make-your-own-crap-database"></a>
</h2>
<p>You can add sequences easily enough to a cRAP FASTA database by just
copying and pasting. However, for reproducibility purposes it is better
to use an R script to generate your own custom cRAP database.</p>
<p>Here we will generate a cRAP database based off of the CCP cRAP, but
with some extra protease sequences added to the end.</p>
<p>First you would download the CCP cRAP FASTA. The following chunk is
not run by default as we have already downloaded it above. In this chunk
we show the “best practice” of naming the CCP cRAP FASTA file with the
current UniProt release.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># download CCP cRAP</span></span>
<span><span class="fu"><a href="../reference/download_ccp_crap.html">download_ccp_crap</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="../reference/check_uniprot_release.html">check_uniprot_release</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"_CCP_cRAP.fasta"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Then you download the sequences you want to add and save them into a
FASTA file. Here we’ll get some protease sequences from the bacteria
<em>Streptomyces griseus</em>.</p>
<p>We can download sequences from UniProt automatically using the
<code><a href="../reference/make_fasta.html">make_fasta()</a></code> function. This function takes a character
vector of UniProt accessions, downloads their sequences, and then saves
the sequences into a FASTA file. We don’t want to add the
<code>cRAP00X</code> numbering to the sequence headers just yet so we
set <code>is_crap = FALSE</code>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">griseus_tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span>fileext <span class="op">=</span> <span class="st">".fasta"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/make_fasta.html">make_fasta</a></span><span class="op">(</span>accessions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"P00776"</span>, <span class="st">"P00777"</span>, <span class="st">"P80561"</span><span class="op">)</span>,</span>
<span>                file <span class="op">=</span> <span class="va">griseus_tmp</span>,</span>
<span>                is_crap <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Downloading from UniProtKB release: 2022_05</span></span></code></pre></div>
<p>Before we add these to our CCP cRAP FASTA, we just need to take note
of what cRAP number the sequences headers stop at.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">tail</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ccp_crap</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "sp|cRAP120|P12763|FETUA_BOVIN Alpha-2-HS-glycoprotein OS=Bos taurus OX=9913 GN=AHSG PE=1 SV=2"                      </span></span>
<span><span class="co">#&gt; [2] "sp|cRAP121|Q58D62|FETUB_BOVIN Fetuin-B OS=Bos taurus OX=9913 GN=FETUB PE=1 SV=1"                                    </span></span>
<span><span class="co">#&gt; [3] "sp|cRAP122|Q28153|CELA1_BOVIN Chymotrypsin-like elastase family member 1 OS=Bos taurus OX=9913 GN=CELA1 PE=2 SV=1"  </span></span>
<span><span class="co">#&gt; [4] "sp|cRAP123|Q29461|CEL2A_BOVIN Chymotrypsin-like elastase family member 2A OS=Bos taurus OX=9913 GN=CELA2A PE=2 SV=1"</span></span>
<span><span class="co">#&gt; [5] "sp|cRAP126|000000|ENDOP_GLUC Endoproteinase GluC NEB P8100S"                                                        </span></span>
<span><span class="co">#&gt; [6] "sp|cRAP127|000000|RECOM_LYSC rLys-C Promega V167A"</span></span></code></pre></div>
<p>The final number is 127, so we want the cRAP numbering of the
additional sequences to start at 128. Finally, we use the
<code><a href="../reference/append_fasta.html">append_fasta()</a></code> function from camprotR to add our <em>S.
griseus</em> sequences to our CCP cRAP FASTA. This function is used to
add the sequences from one FASTA (file1) onto the end of another FASTA
(file2).</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/append_fasta.html">append_fasta</a></span><span class="op">(</span></span>
<span>  file1 <span class="op">=</span> <span class="va">griseus_tmp</span>,</span>
<span>  file2 <span class="op">=</span> <span class="va">ccp_tmp</span>,</span>
<span>  is_crap <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  crap_start <span class="op">=</span> <span class="fl">128</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Lets just check that this worked properly. There should be 130
sequences now.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">Biostrings</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html" class="external-link">readAAStringSet</a></span><span class="op">(</span><span class="va">ccp_tmp</span><span class="op">)</span></span>
<span><span class="co">#&gt; AAStringSet object of length 128:</span></span>
<span><span class="co">#&gt;       width seq                                             names               </span></span>
<span><span class="co">#&gt;   [1]   348 MSIPETQKGVIFYESHGKLEYK...PEIYEKMEKGQIVGRYVVDTSK sp|cRAP001|P00330...</span></span>
<span><span class="co">#&gt;   [2]   609 MKWVTFISLLFLFSSAYSRGVF...KETCFAEEGKKLVAASQAALGL sp|cRAP002|P02768...</span></span>
<span><span class="co">#&gt;   [3]   364 MPHSHPALTPEQKKELSDIAHR...YTPSGQAGAAASESLFISNHAY sp|cRAP003|P00883...</span></span>
<span><span class="co">#&gt;   [4]   511 MKLFWLLFTIGFCWAQYSSNTQ...AHFSISNSAEDPFIAIHAESKL sp|cRAP004|P0DUB6...</span></span>
<span><span class="co">#&gt;   [5]   511 MKLFWLLFTIGFCWAQYSSNTQ...AHFSISNSAEDPFIAIHAESKL sp|cRAP005|P0DTE7...</span></span>
<span><span class="co">#&gt;   ...   ... ...</span></span>
<span><span class="co">#&gt; [124]   274 VILPNNDRHQITDTTNGHYAPV...NPDNGDNNNSDNPDAAHHHHHH sp|cRAP126|000000...</span></span>
<span><span class="co">#&gt; [125]   261 AGYRDGFGASGSCEVDAVCATQ...GVYSQISRYFAPHQHQHQHQHQ sp|cRAP127|000000...</span></span>
<span><span class="co">#&gt; [126]   297 MTFKRFSPLSSTSRYARLLAVA...TGGTTFYQPVTEALSAYGATVL sp|cRAP128|P00776...</span></span>
<span><span class="co">#&gt; [127]   299 MRIKRTSNRSNAARRVRTTAVL...SGGTTFFQPVTEALSAYGVSVY sp|cRAP129|P00777...</span></span>
<span><span class="co">#&gt; [128]   445 MRPNRFSLRRSPTAVAAVALAA...KLRVQDIARQDTGYIDSWKLTF sp|cRAP130|P80561...</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Charlotte Dawson, Tom Smith.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
