<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>LFQ QC and summarisation to protein-level abundance • camprotR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="LFQ QC and summarisation to protein-level abundance">
<meta property="og:description" content="camprotR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">camprotR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/LFQ_parse_qc_protein_inference.html">LFQ QC and summarisation to protein-level abundance</a>
    </li>
    <li>
      <a href="../articles/SILAC.html">Processing and QC of SILAC data</a>
    </li>
    <li>
      <a href="../articles/TMT_PSM_QC_Summarisation.html">TMT QC PSM-level quantification and summarisation to protein-level abundance</a>
    </li>
    <li>
      <a href="../articles/crap.html">cRAP databases</a>
    </li>
    <li>
      <a href="../articles/msnset.html">Working with MSnSets</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>LFQ QC and summarisation to protein-level
abundance</h1>
                        <h4 data-toc-skip class="author">Tom Smith</h4>
            
            <h4 data-toc-skip class="date">2023-01-10</h4>
      
      
      <div class="hidden name"><code>LFQ_parse_qc_protein_inference.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">camprotR</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biostrings" class="external-link">Biostrings</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://lgatto.github.io/MSnbase" class="external-link">MSnbase</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/njtierney/naniar" class="external-link">naniar</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/talgalili/gplots" class="external-link">gplots</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/StoreyLab/biobroom" class="external-link">biobroom</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://httr.r-lib.org/" class="external-link">httr</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="input-data">Input data<a class="anchor" aria-label="anchor" href="#input-data"></a>
</h2>
<p>We start by reading in a file containing peptide-level output from
Proteome Discoverer (PD). This experiment was designed to identify
RNA-binding proteins (RBPs) in the U-2 OS cell line using the OOPS
method <span class="citation">(Queiroz et al. 2019)</span> with a
comparison of RNase +/- used to separate RBPs from background
non-specific proteins. 4 replicate experiments were performed, where the
RNase +/- experiments were performed from the same OOPS interface
resulting in 8 samples in total. For each LFQ run, approximately the
same quantity of protein was injected, based on quantification of
peptide concentration post trypsin digestion. This data is not published
but aim of the experiment is equivalent to figure 2e in the original
OOPS paper <span class="citation">(Queiroz et al. 2019)</span>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pep_data</span> <span class="op">&lt;-</span> <span class="fu">camprotR</span><span class="fu">::</span><span class="va"><a href="../reference/lfq_oops_rnase_pep.html">lfq_oops_rnase_pep</a></span></span></code></pre></div>
<p>The samples analysed are as follows:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sample_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  File <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"F"</span>, <span class="fl">17</span><span class="op">:</span><span class="fl">24</span><span class="op">)</span>,</span>
<span>  Sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"RNase_neg"</span>, <span class="st">"RNase_pos"</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">4</span><span class="op">)</span>, <span class="st">"."</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span>,</span>
<span>  Condition <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"RNase_neg"</span>, <span class="st">"RNase_pos"</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">4</span><span class="op">)</span>,</span>
<span>  Replicate <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span>, <span class="fl">2</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">sample_data</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">File</th>
<th align="left">Sample</th>
<th align="left">Condition</th>
<th align="right">Replicate</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">F17</td>
<td align="left">RNase_neg.1</td>
<td align="left">RNase_neg</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">F18</td>
<td align="left">RNase_neg.2</td>
<td align="left">RNase_neg</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">F19</td>
<td align="left">RNase_neg.3</td>
<td align="left">RNase_neg</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="left">F20</td>
<td align="left">RNase_neg.4</td>
<td align="left">RNase_neg</td>
<td align="right">4</td>
</tr>
<tr class="odd">
<td align="left">F21</td>
<td align="left">RNase_pos.1</td>
<td align="left">RNase_pos</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">F22</td>
<td align="left">RNase_pos.2</td>
<td align="left">RNase_pos</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">F23</td>
<td align="left">RNase_pos.3</td>
<td align="left">RNase_pos</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="left">F24</td>
<td align="left">RNase_pos.4</td>
<td align="left">RNase_pos</td>
<td align="right">4</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="parse-input-data-and-remove-contaminants">Parse input data and remove contaminants<a class="anchor" aria-label="anchor" href="#parse-input-data-and-remove-contaminants"></a>
</h2>
<p>The first step is to remove contaminant proteins. These were defined
using the <a href="https://www.thegpm.org/crap/" class="external-link">cRAP database</a>.
Below, we parse the cRAP FASTA to extract the IDs for the cRAP proteins,
in both ‘cRAP’ format and UniProt accessions for these proteins.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">crap_fasta_inf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span></span>
<span>  <span class="st">"extdata"</span>, <span class="st">"cRAP_20190401.fasta.gz"</span>, </span>
<span>  package <span class="op">=</span> <span class="st">"camprotR"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load the cRAP FASTA used for the PD search</span></span>
<span><span class="va">crap_fasta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html" class="external-link">fasta.index</a></span><span class="op">(</span><span class="va">crap_fasta_inf</span>, seqtype <span class="op">=</span> <span class="st">"AA"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Define a base R version of stringr::str_extract_all()</span></span>
<span><span class="va">str_extract_all</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">pattern</span>, <span class="va">string</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gregexpr</a></span><span class="op">(</span><span class="va">pattern</span>, <span class="va">string</span>, perl <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/regmatches.html" class="external-link">regmatches</a></span><span class="op">(</span><span class="va">string</span>, <span class="va">.</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Extract the UniProt accessions associated with each cRAP protein</span></span>
<span><span class="va">crap_accessions</span> <span class="op">&lt;-</span> <span class="va">crap_fasta</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">desc</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">str_extract_all</span><span class="op">(</span><span class="st">"(?&lt;=\\|).*?(?=\\|)"</span>, <span class="va">.</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>We can then supply these cRAP protein IDs to
<code><a href="../reference/parse_features.html">parse_features()</a></code> which will remove features (i.e. peptides
in this case) which may originate from contaminants, as well as features
which don’t have a unique master protein.</p>
<p>See <code><a href="../reference/parse_features.html">?parse_features</a></code> for further details, including the
removal of ‘associated cRAP’ for conservative contaminants removal.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pep_data_flt</span> <span class="op">&lt;-</span> <span class="fu">camprotR</span><span class="fu">::</span><span class="fu"><a href="../reference/parse_features.html">parse_features</a></span><span class="op">(</span></span>
<span>  <span class="va">pep_data</span>,</span>
<span>  TMT <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>  level <span class="op">=</span> <span class="st">'peptide'</span>,</span>
<span>  crap_proteins <span class="op">=</span> <span class="va">crap_accessions</span>, </span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Parsing features...</span></span>
<span><span class="co">#&gt; 496 features found from 100 master proteins =&gt; Input</span></span>
<span><span class="co">#&gt; 242 cRAP proteins supplied</span></span>
<span><span class="co">#&gt; 28 proteins identified as 'cRAP associated'</span></span>
<span><span class="co">#&gt; 440 features found from 96 master proteins =&gt; cRAP features removed</span></span>
<span><span class="co">#&gt; 440 features found from 96 master proteins =&gt; associated cRAP features removed</span></span>
<span><span class="co">#&gt; 410 features found from 84 master proteins =&gt; features with non-unique master proteins removed</span></span></code></pre></div>
<p>From the above, we can see that we have started with 496 ‘features’
(peptides) from 100 master proteins across all samples. After removal of
contaminants and peptides that can’t be assigned to a unique master
protein, we have 410 peptides remaining from 84 master proteins.</p>
<p>We now store the filtered peptide data in an MSnSet, the standard
data object for proteomics in R. See the <code><a href="../articles/msnset.html">vignette("msnset")</a></code>
for more details.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create expression matrix with peptide abundances (exprs) and human </span></span>
<span><span class="co"># readable column names</span></span>
<span><span class="va">exprs_data</span> <span class="op">&lt;-</span> <span class="va">pep_data_flt</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">matches</a></span><span class="op">(</span><span class="st">"Abundance"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">`colnames&lt;-`</span><span class="op">(</span><span class="va">sample_data</span><span class="op">$</span><span class="va">Sample</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create data.frame with sample metadata (pData)</span></span>
<span><span class="va">pheno_data</span> <span class="op">&lt;-</span> <span class="va">sample_data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">File</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"Sample"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create data.frame with peptide metadata (fData)</span></span>
<span><span class="va">feat_data</span> <span class="op">&lt;-</span> <span class="va">pep_data_flt</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">matches</a></span><span class="op">(</span><span class="st">"Abundance"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create MSnSet</span></span>
<span><span class="va">pep</span> <span class="op">&lt;-</span> <span class="fu">MSnbase</span><span class="fu">::</span><span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/MSnSet-class.html" class="external-link">MSnSet</a></span><span class="op">(</span>exprs <span class="op">=</span> <span class="va">exprs_data</span>,</span>
<span>                       fData <span class="op">=</span> <span class="va">feat_data</span>,</span>
<span>                       pData <span class="op">=</span> <span class="va">pheno_data</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="qc-peptides">QC peptides<a class="anchor" aria-label="anchor" href="#qc-peptides"></a>
</h2>
<p>First of all, we want to inspect the peptide intensity distributions.
We expect these to be approximately equal and any very low intensity
sample would be a concern that would need to be further explored. Here,
we can see that there is some clear variability, but no sample with very
low intensity.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pep</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span>base <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">camprotR</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_quant.html">plot_quant</a></span><span class="op">(</span>method <span class="op">=</span> <span class="st">'density'</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 1169 rows containing non-finite values</span></span>
<span><span class="co">#&gt; (`stat_density()`).</span></span></code></pre></div>
<div class="figure">
<img src="LFQ_parse_qc_protein_inference_files/figure-html/unnamed-chunk-7-1.png" alt="Peptide intensities" width="384"><p class="caption">
Peptide intensities
</p>
</div>
<p>Next, we consider the missing values. Note that
<code><a href="https://lgatto.github.io/MSnbase/reference/plotNA-methods.html" class="external-link">MSnbase::plotNA</a></code> assumes the object contains protein-level
data and names the x-axis accordingly. Here, we update the plot
aesthetics and rename the x-axis</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">MSnbase</span><span class="fu">::</span><span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/plotNA-methods.html" class="external-link">plotNA</a></span><span class="op">(</span><span class="va">pep</span>, pNA <span class="op">=</span> <span class="fl">0</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">camprotR</span><span class="fu">::</span><span class="fu"><a href="../reference/theme_camprot.html">theme_camprot</a></span><span class="op">(</span>border <span class="op">=</span> <span class="cn">FALSE</span>, base_family <span class="op">=</span> <span class="st">'sans'</span>, base_size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">'Peptide index'</span><span class="op">)</span></span></code></pre></div>
<p>So, from the 410 peptides, just 107 have quantification values in all
8 samples. This is not a suprise for LFQ, since each sample is prepared
and run separately.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="LFQ_parse_qc_protein_inference_files/figure-html/unnamed-chunk-9-1.png" alt="Data completeness, all peptides" width="480"><p class="caption">
Data completeness, all peptides
</p>
</div>
<p>We can explore the structure of the missing values further using an
‘upset’ plot.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">missing_data</span> <span class="op">&lt;-</span> <span class="va">pep</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">naniar</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/naniar/man/gg_miss_upset.html" class="external-link">gg_miss_upset</a></span><span class="op">(</span><span class="va">missing_data</span>,</span>
<span>                      sets <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pep</span><span class="op">)</span>, <span class="st">'_NA'</span><span class="op">)</span>,</span>
<span>                      keep.order <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                      nsets <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="LFQ_parse_qc_protein_inference_files/figure-html/unnamed-chunk-10-1.png" alt="Missing values upset plot" width="768"><p class="caption">
Missing values upset plot
</p>
</div>
<p>So in this case, we can see that the most common missing value
patterns are:</p>
<ol style="list-style-type: decimal">
<li>Missing in just RNase negative replicate 4</li>
<li>Missing in all samples.</li>
<li>Missing in all the other samples, except RNase negative replicate
4</li>
</ol>
<p>RNase negative replicate 4 had slightly lower overall petide
intensities and appears to be somewhat of an outlier. In this case, we
will retain the sample but in other cases, this may warrant further
exploration and potentially removal of a sample.</p>
<p>Finally, we recode the abundances to binary (present/absent) and plot
the missing data structure as a heatmap (missing = black). Beyond the
aforementioned observations regarding RNase negative replicate 4,
there’s no clear structure in the missing values. This is what we would
expect for LFQ, where data is largely missing at random (MAR). There are
methods to impute values MAR, but, providing what we are after is
protein-level abundance, it is usually more appropriate to perform the
protein inference using a method that can accommodate missing
values.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">missing_data</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">missing_data</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">1</span></span>
<span><span class="va">missing_data</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">missing_data</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">0</span></span>
<span></span>
<span><span class="fu">gplots</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gplots/man/heatmap.2.html" class="external-link">heatmap.2</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">missing_data</span><span class="op">)</span>,</span>
<span>  col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"black"</span>, <span class="st">"lightgray"</span><span class="op">)</span>,</span>
<span>  scale <span class="op">=</span> <span class="st">"none"</span>, <span class="co"># Don't re-scale the input</span></span>
<span>  <span class="co"># Default is euclidean, which is less appropriate distance for binary vectors</span></span>
<span>  distfun <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/stats/dist.html" class="external-link">dist</a></span><span class="op">(</span><span class="va">x</span>, method <span class="op">=</span> <span class="st">'binary'</span><span class="op">)</span>, </span>
<span>  trace <span class="op">=</span> <span class="st">"none"</span>, <span class="co"># Don't include a trace. Improves appearance</span></span>
<span>  key <span class="op">=</span> <span class="cn">FALSE</span>, <span class="co"># Don't include a key as values are binary</span></span>
<span>  Colv <span class="op">=</span> <span class="cn">FALSE</span>, <span class="co"># Don't re-order the columns</span></span>
<span>  labRow <span class="op">=</span> <span class="cn">FALSE</span>, <span class="co"># Don't label the rows</span></span>
<span>  cexCol <span class="op">=</span> <span class="fl">0.7</span> <span class="co"># Reduce the column name size</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in gplots::heatmap.2(as.matrix(missing_data), col = c("black",</span></span>
<span><span class="co">#&gt; "lightgray"), : Discrepancy: Colv is FALSE, while dendrogram is `both'. Omitting</span></span>
<span><span class="co">#&gt; column dendogram.</span></span></code></pre></div>
<div class="figure">
<img src="LFQ_parse_qc_protein_inference_files/figure-html/unnamed-chunk-11-1.png" alt="Presence/Absence heatmap" width="576"><p class="caption">
Presence/Absence heatmap
</p>
</div>
</div>
<div class="section level2">
<h2 id="normalise-peptide-intensities">Normalise peptide intensities<a class="anchor" aria-label="anchor" href="#normalise-peptide-intensities"></a>
</h2>
<p>We have injected the same quantity of peptides for each sample, so
it’s reasonable to normalise the samples against one another. Here we
will apply ‘center-median’ normalisation, which in
<code><a href="https://lgatto.github.io/MSnbase/reference/normalise-methods.html" class="external-link">MSnbase::normalise</a></code> is called ‘diff.median’. Since the
peptide intensities are log-Gaussian distributed, we
log<sub>2</sub>-transform them before performing the normalisation.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pep_norm</span> <span class="op">&lt;-</span> <span class="va">pep</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span>base <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">MSnbase</span><span class="fu">::</span><span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/normalise-methods.html" class="external-link">normalise</a></span><span class="op">(</span><span class="st">'diff.median'</span><span class="op">)</span>  </span>
<span></span>
<span><span class="va">pep_norm</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">camprotR</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_quant.html">plot_quant</a></span><span class="op">(</span>method <span class="op">=</span> <span class="st">'density'</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 1169 rows containing non-finite values</span></span>
<span><span class="co">#&gt; (`stat_density()`).</span></span></code></pre></div>
<div class="figure">
<img src="LFQ_parse_qc_protein_inference_files/figure-html/unnamed-chunk-12-1.png" alt="Protein intensities post-normalisation" width="480"><p class="caption">
Protein intensities post-normalisation
</p>
</div>
</div>
<div class="section level2">
<h2 id="summarising-to-protein-level-abundance">Summarising to protein-level abundance<a class="anchor" aria-label="anchor" href="#summarising-to-protein-level-abundance"></a>
</h2>
<p>Before we can summarise to protein-level abundances, we need to
exclude peptides with too many missing values. Here, peptides with more
than 5/8 missing values are discarded, using
<code><a href="https://rdrr.io/pkg/ProtGenerics/man/protgenerics.html" class="external-link">MSnbase::filterNA()</a></code>. We also need to remove proteins
without at least two peptides. We will use
<code><a href="../reference/restrict_features_per_protein.html">camprotR::restrict_features_per_protein()</a></code> which will
replace quantification values with <code>NA</code> if the sample does
not have two quantified peptides for a given protein. Note that this
means we have to repeat the filtering process since we are adding
missing values.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pep_restricted</span> <span class="op">&lt;-</span> <span class="va">pep_norm</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">MSnbase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ProtGenerics/man/protgenerics.html" class="external-link">filterNA</a></span><span class="op">(</span>pNA <span class="op">=</span> <span class="fl">4</span><span class="op">/</span><span class="fl">8</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="co"># Maximum 4/8 missing values</span></span>
<span>  <span class="fu">camprotR</span><span class="fu">::</span><span class="fu"><a href="../reference/restrict_features_per_protein.html">restrict_features_per_protein</a></span><span class="op">(</span>min_features <span class="op">=</span> <span class="fl">2</span>, plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="co"># At least two peptides per protein</span></span>
<span>  <span class="co"># Repeat the filtering since restrict_features_per_protein will replace some values with NA</span></span>
<span>  <span class="fu">MSnbase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ProtGenerics/man/protgenerics.html" class="external-link">filterNA</a></span><span class="op">(</span>pNA <span class="op">=</span> <span class="fl">4</span><span class="op">/</span><span class="fl">8</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">camprotR</span><span class="fu">::</span><span class="fu"><a href="../reference/restrict_features_per_protein.html">restrict_features_per_protein</a></span><span class="op">(</span>min_features <span class="op">=</span> <span class="fl">2</span>, plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> </span></code></pre></div>
<p>We can then re-inspect the missing values. Note that we have reduced
the overall number of peptides to 271.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">MSnbase</span><span class="fu">::</span><span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/plotNA-methods.html" class="external-link">plotNA</a></span><span class="op">(</span><span class="va">pep_restricted</span>, pNA <span class="op">=</span> <span class="fl">0</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">camprotR</span><span class="fu">::</span><span class="fu"><a href="../reference/theme_camprot.html">theme_camprot</a></span><span class="op">(</span>border <span class="op">=</span> <span class="cn">FALSE</span>, base_family <span class="op">=</span> <span class="st">'sans'</span>, base_size <span class="op">=</span> <span class="fl">15</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">'Peptide index'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="LFQ_parse_qc_protein_inference_files/figure-html/unnamed-chunk-15-1.png" alt="Data completeness, retained peptides" width="576"><p class="caption">
Data completeness, retained peptides
</p>
</div>
<p>We can now summarise to protein-level abundance. Below, we use
‘robust’ summarisation <span class="citation">(Sticker et al.
2020)</span> with <code><a href="https://rdrr.io/pkg/ProtGenerics/man/protgenerics.html" class="external-link">MSnbase::combineFeatures()</a></code>. This returns
a warning about missing values that we can ignore here since the robust
method is inherently designed to handle missing values. See
<code><a href="https://rdrr.io/pkg/MsCoreUtils/man/robustSummary.html" class="external-link">MsCoreUtils::robustSummary()</a></code> and <a href="https://pubmed.ncbi.nlm.nih.gov/32321741/" class="external-link">this publication</a>
for further details about the robust method.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">prot_robust</span> <span class="op">&lt;-</span> <span class="va">pep_restricted</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">MSnbase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ProtGenerics/man/protgenerics.html" class="external-link">combineFeatures</a></span><span class="op">(</span></span>
<span>    <span class="co"># group the peptides by their master protein id</span></span>
<span>    groupBy <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">pep_restricted</span><span class="op">)</span><span class="op">$</span><span class="va">Master.Protein.Accessions</span>, </span>
<span>    method <span class="op">=</span> <span class="st">'robust'</span>, </span>
<span>    maxit <span class="op">=</span> <span class="fl">1000</span>  <span class="co"># Ensures convergence for MASS::rlm</span></span>
<span>  <span class="op">)</span> </span>
<span><span class="co">#&gt; Your data contains missing values. Please read the relevant section in</span></span>
<span><span class="co">#&gt; the combineFeatures manual page for details on the effects of missing</span></span>
<span><span class="co">#&gt; values on data aggregation.</span></span></code></pre></div>
<p>We can then re-inspect the missing values at the protein level. So,
we have quantification for 46 proteins, of which 26 are fully quantified
across all 8 samples. The most common missing values pattern remains
missing in just RNase negative replicate 4.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">MSnbase</span><span class="fu">::</span><span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/plotNA-methods.html" class="external-link">plotNA</a></span><span class="op">(</span><span class="va">prot_robust</span>, pNA <span class="op">=</span> <span class="fl">0</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">camprotR</span><span class="fu">::</span><span class="fu"><a href="../reference/theme_camprot.html">theme_camprot</a></span><span class="op">(</span>border <span class="op">=</span> <span class="cn">FALSE</span>, base_family <span class="op">=</span> <span class="st">'sans'</span>, base_size <span class="op">=</span> <span class="fl">15</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></span></code></pre></div>
<p><img src="LFQ_parse_qc_protein_inference_files/figure-html/unnamed-chunk-18-1.png" width="576"></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">naniar</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/naniar/man/gg_miss_upset.html" class="external-link">gg_miss_upset</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">prot_robust</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                      sets <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">prot_robust</span><span class="op">)</span>, <span class="st">'_NA'</span><span class="op">)</span>,</span>
<span>                      keep.order <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                      nsets <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<p><img src="LFQ_parse_qc_protein_inference_files/figure-html/unnamed-chunk-19-1.png" width="768"></p>
<p>We have now obtained protein-level abundances from our LFQ data. This
is our recommended procedure for summarising from peptide to protein
level abundances.</p>
</div>
<div class="section level2">
<h2 id="alternatives-for-summarising-to-protein-level-abundance---maxlfq">Alternatives for summarising to protein-level abundance -
MaxLFQ<a class="anchor" aria-label="anchor" href="#alternatives-for-summarising-to-protein-level-abundance---maxlfq"></a>
</h2>
<p>Although we recommend using the ‘robust’ summarisation, there are
other algorithms that are commonly applied. For example MaxLFQ <span class="citation">(Cox et al. 2014)</span>, which is implemented within
MaxQuant, but also available via the <code><a href="https://rdrr.io/pkg/iq/man/maxLFQ.html" class="external-link">iq::maxLFQ()</a></code> function
in the iq package in R (now available on <a href="https://cran.r-project.org/web/packages/iq/index.html" class="external-link">CRAN</a>).</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># You may wish to retain more feature columns that this! </span></span>
<span><span class="va">feature_coloumns_to_retain</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Master.Protein.Accessions"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># To ensure we are only comparing the protein inference method and not normalisation etc,</span></span>
<span><span class="co"># we will use the same set of peptides for both inference methods</span></span>
<span><span class="va">pep_data_for_summarisation</span> <span class="op">&lt;-</span> <span class="va">pep_restricted</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">pep_restricted</span><span class="op">)</span><span class="op">[</span>, <span class="va">feature_coloumns_to_retain</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>        by <span class="op">=</span> <span class="st">'row.names'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">Row.names</span><span class="op">)</span></span></code></pre></div>
<p>Below, we use <code><a href="https://rdrr.io/pkg/iq/man/maxLFQ.html" class="external-link">iq::maxLFQ()</a></code> manually, though it should be
possible to use this function within
<code><a href="https://rdrr.io/pkg/ProtGenerics/man/protgenerics.html" class="external-link">MSnbase::combineFeatures()</a></code> too in theory since it
accommodates user-defined functions.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Define a function to perform MaxLFQ on a single protein and return a data.frame</span></span>
<span><span class="co"># as required. </span></span>
<span><span class="va">get_maxlfq_estimate</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">obj</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">prot</span> <span class="op">&lt;-</span> <span class="fu">iq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/iq/man/maxLFQ.html" class="external-link">maxLFQ</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">estimate</span></span>
<span>  </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">prot</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Group by the features we want to retain and use MaxLFQ on each protein</span></span>
<span><span class="va">maxlfq_estimates</span> <span class="op">&lt;-</span> <span class="va">pep_data_for_summarisation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/across.html" class="external-link">across</a></span><span class="op">(</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/all_of.html" class="external-link">all_of</a></span><span class="op">(</span><span class="va">feature_coloumns_to_retain</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_map.html" class="external-link">group_modify</a></span><span class="op">(</span><span class="op">~</span> <span class="fu">get_maxlfq_estimate</span><span class="op">(</span><span class="va">.</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">ungroup</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create the protein-level MSnSet</span></span>
<span><span class="va">maxlfq.e</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">maxlfq_estimates</span>, <span class="op">-</span><span class="va">Master.Protein.Accessions</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">maxlfq.f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">maxlfq_estimates</span>, <span class="va">Master.Protein.Accessions</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">maxlfq.p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">pep_restricted</span><span class="op">)</span></span>
<span></span>
<span><span class="va">prot_maxlfq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/MSnSet-class.html" class="external-link">MSnSet</a></span><span class="op">(</span>exprs <span class="op">=</span> <span class="va">maxlfq.e</span>,</span>
<span>                      fData <span class="op">=</span> <span class="va">maxlfq.f</span>,</span>
<span>                      pData <span class="op">=</span> <span class="va">maxlfq.p</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Update the rownames to be the protein IDs</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">prot_maxlfq</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">maxlfq_estimates</span><span class="op">$</span><span class="va">Master.Protein.Accessions</span></span></code></pre></div>
<p>We can now compare the protein-level abundance estimates.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Define a function to extract the protein abundances in long form and</span></span>
<span><span class="co"># add a column annotating the method</span></span>
<span><span class="va">get_long_form_prot_exp</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">obj</span>, <span class="va">method_name</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://generics.r-lib.org/reference/tidy.html" class="external-link">tidy</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>method <span class="op">=</span> <span class="va">method_name</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Single object with protein inference from both methods </span></span>
<span><span class="va">compare_protein_abundances</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span></span>
<span>  <span class="fu">get_long_form_prot_exp</span><span class="op">(</span><span class="va">prot_maxlfq</span>, <span class="st">'MaxLFQ'</span><span class="op">)</span>,</span>
<span>  <span class="fu">get_long_form_prot_exp</span><span class="op">(</span><span class="va">prot_robust</span>, <span class="st">'Robust'</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot direct comparison</span></span>
<span><span class="va">compare_protein_abundances</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_wider.html" class="external-link">pivot_wider</a></span><span class="op">(</span>names_from <span class="op">=</span> <span class="va">method</span>, values_from <span class="op">=</span> <span class="va">value</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="co"># pivot to wider form for plotting</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Robust</span>, y <span class="op">=</span> <span class="va">MaxLFQ</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.25</span>, size <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/theme_camprot.html">theme_camprot</a></span><span class="op">(</span>border <span class="op">=</span> <span class="cn">FALSE</span>, base_family <span class="op">=</span> <span class="st">'sans'</span>, base_size <span class="op">=</span> <span class="fl">15</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_abline</a></span><span class="op">(</span>slope <span class="op">=</span> <span class="fl">1</span>, colour <span class="op">=</span> <span class="fu"><a href="../reference/get_cat_palette.html">get_cat_palette</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, linetype <span class="op">=</span> <span class="fl">2</span>, size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="LFQ_parse_qc_protein_inference_files/figure-html/unnamed-chunk-22-1.png" alt="Comparison of protein inference methods" width="480"><p class="caption">
Comparison of protein inference methods
</p>
</div>
<p>There is a very good overall correlation. Let’s inspect a few
proteins with the largest differences between the two approaches to see
what’s going on for the edge cases.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Identify proteins with largest difference between the protein summarisation methods</span></span>
<span><span class="va">proteins_of_interest</span> <span class="op">&lt;-</span> <span class="va">compare_protein_abundances</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_wider.html" class="external-link">pivot_wider</a></span><span class="op">(</span>names_from <span class="op">=</span> <span class="va">method</span>, values_from <span class="op">=</span> <span class="va">value</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>diff <span class="op">=</span> <span class="va">MaxLFQ</span><span class="op">-</span><span class="va">Robust</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">diff</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">protein</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fl">5</span><span class="op">)</span></span></code></pre></div>
<p>Below we define a function to plot the peptide and protein abundances
for the two methods for a single protein. We can ignore the details
since it’s the plots themselves we are interested in.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot_pep_and_protein</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">protein_of_interest</span><span class="op">)</span> <span class="op">{</span></span>
<span>  </span>
<span>  <span class="va">to_plot_compare</span> <span class="op">&lt;-</span> <span class="va">compare_protein_abundances</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">protein</span> <span class="op">==</span> <span class="va">protein_of_interest</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="va">pep_restricted</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">pep_restricted</span><span class="op">)</span><span class="op">$</span><span class="va">Master.Protein.Accession</span> <span class="op">==</span> <span class="va">protein_of_interest</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span><span class="st">'id'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span>cols <span class="op">=</span> <span class="op">-</span><span class="va">id</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">name</span>, y <span class="op">=</span> <span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_line</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>group <span class="op">=</span> <span class="va">id</span><span class="op">)</span>, colour <span class="op">=</span> <span class="st">'grey'</span>, alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>colour <span class="op">=</span> <span class="st">'grey'</span>, alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_line</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">to_plot_compare</span>,</span>
<span>              <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">sample.id</span>, y <span class="op">=</span> <span class="va">value</span>, colour <span class="op">=</span> <span class="va">method</span>, group <span class="op">=</span> <span class="va">method</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">to_plot_compare</span>,</span>
<span>               <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">sample.id</span>, y <span class="op">=</span> <span class="va">value</span>, colour <span class="op">=</span> <span class="va">method</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_colour_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="../reference/get_cat_palette.html">get_cat_palette</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span>, name <span class="op">=</span> <span class="st">'LFQ summarisation method'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="../reference/theme_camprot.html">theme_camprot</a></span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">15</span>, border <span class="op">=</span> <span class="cn">FALSE</span>, base_family <span class="op">=</span> <span class="st">'sans'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, vjust <span class="op">=</span> <span class="fl">1</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>      title <span class="op">=</span> <span class="va">protein_of_interest</span>,</span>
<span>      x <span class="op">=</span> <span class="st">''</span>,</span>
<span>      y <span class="op">=</span> <span class="st">'Protein abundance (log2)'</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p>Below we apply the above function to each of our proteins of
interest.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">proteins_of_interest</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">plot_pep_and_protein</span><span class="op">)</span></span>
<span><span class="co">#&gt; [[1]]</span></span></code></pre></div>
<p><img src="LFQ_parse_qc_protein_inference_files/figure-html/unnamed-chunk-25-1.png" width="672"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[2]]</span></span></code></pre>
<p><img src="LFQ_parse_qc_protein_inference_files/figure-html/unnamed-chunk-25-2.png" width="672"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[3]]</span></span></code></pre>
<p><img src="LFQ_parse_qc_protein_inference_files/figure-html/unnamed-chunk-25-3.png" width="672"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[4]]</span></span></code></pre>
<p><img src="LFQ_parse_qc_protein_inference_files/figure-html/unnamed-chunk-25-4.png" width="672"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[5]]</span></span></code></pre>
<p><img src="LFQ_parse_qc_protein_inference_files/figure-html/unnamed-chunk-25-5.png" width="672"></p>
<p>Looking at these examples, we can see that MaxLFQ is often estimating
slightly higher abundances but with a very similar profile across the
samples is very similar, so the summarisation approach is unlikely to
affect the downstream analysis. It’s not clear which of the two
approaches is more correct in the examples above, but the publication
proposing the robust protein inference (see <a href="https://pubmed.ncbi.nlm.nih.gov/32321741/" class="external-link">here</a>) does indicate
it gives more accurate fold-change estimates overall.</p>
<p>We have now processed our peptide-level LFQ abundances and obtained
protein-level abundances, from which we can perform our downstream
analyses.</p>
</div>
<div class="section level2">
<h2 id="alternative-normalisation---using-reference-proteins">Alternative normalisation - Using reference proteins<a class="anchor" aria-label="anchor" href="#alternative-normalisation---using-reference-proteins"></a>
</h2>
<p>Finally, we consider an alternative normalisation approach where we
have a strong prior expectation about the abundance of a subset of
proteins. Our data is from an RNase +/- OOPS experiment. Since OOPS is
known to enrich glycoproteins at the interface, we can use these
proteins as an internal set of ‘housekeeping’ proteins which we expect
to have no difference in abundance between RNase +/-.</p>
<p>First, we need to calculate the RNase +/- ratios, since this is the
value we want to normalise. In other applications, you may wish to
normalise the abundance estimates for each sample, rather than the ratio
between samples.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ratios</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">prot_robust</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">prot_robust</span><span class="op">)</span><span class="op">$</span><span class="va">Condition</span> <span class="op">==</span> <span class="st">'RNase_neg'</span><span class="op">]</span><span class="op">)</span> <span class="op">-</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">prot_robust</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">prot_robust</span><span class="op">)</span><span class="op">$</span><span class="va">Condition</span> <span class="op">==</span> <span class="st">'RNase_pos'</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="va">prot_ratios</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/MSnSet-class.html" class="external-link">MSnSet</a></span><span class="op">(</span>exprs <span class="op">=</span> <span class="va">ratios</span>,</span>
<span>                      fData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">prot_robust</span><span class="op">)</span>,</span>
<span>                      pData <span class="op">=</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">prot_robust</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Condition</span> <span class="op">==</span> <span class="st">'RNase_neg'</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Next, we need annotations regarding which proteins are glycoproteins
and GO-annotated RBPs. [Once a notebook has been added to cover GO
annotations, reference to that here to avoid repeating explanations
about why we want to expand the set of GO anntotations]</p>
<p>Get all GO terms for our proteins by querying UniProt
programmatically.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># remotes::install_github("csdaw/uniprotREST")</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">uniprotREST</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/uniprotREST/man/uniprot_map.html" class="external-link">uniprot_map</a></span><span class="op">(</span></span>
<span>  ids <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">prot_robust</span><span class="op">)</span>,</span>
<span>  fields <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"accession"</span>, <span class="st">"feature_count"</span>, <span class="st">"go_id"</span><span class="op">)</span>,</span>
<span>  format <span class="op">=</span> <span class="st">"tsv"</span>,</span>
<span>  verbosity <span class="op">=</span> <span class="fl">0</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="va">res</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Entry</span>, <span class="va">Features</span>, <span class="va">Gene.Ontology.IDs</span><span class="op">)</span></span></code></pre></div>
<p>Get RBPs</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># GO term for RNA-binding</span></span>
<span><span class="va">go_rbp</span> <span class="op">&lt;-</span> <span class="st">"GO:0003723"</span></span>
<span></span>
<span><span class="co"># All offspring (since some proteins will only be annotated with an offspring term)</span></span>
<span><span class="va">go_rbp_offspring</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html" class="external-link">get</a></span><span class="op">(</span><span class="st">"GO:0003723"</span>, <span class="fu">GO.db</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/GO.db/man/GOMFOFFSPRING.html" class="external-link">GOMFOFFSPRING</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span></span>
<span><span class="co"># Identify all GO-RBPs</span></span>
<span><span class="va">rbps</span><span class="op">&lt;-</span> <span class="va">res</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/separate_rows.html" class="external-link">separate_rows</a></span><span class="op">(</span><span class="va">Gene.Ontology.IDs</span>, sep<span class="op">=</span><span class="st">'; '</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Gene.Ontology.IDs</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">go_rbp</span>, <span class="va">go_rbp_offspring</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">Entry</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get glycoproteins</span></span>
<span><span class="va">glycoproteins</span> <span class="op">&lt;-</span> <span class="va">res</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"Glycosylation"</span>, <span class="va">Features</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Glyco.features <span class="op">=</span> <span class="va">Features</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/separate_rows.html" class="external-link">separate_rows</a></span><span class="op">(</span><span class="va">Glyco.features</span>, sep <span class="op">=</span> <span class="st">"; "</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"Glycosylation"</span>, <span class="va">Glyco.features</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Glyco.features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"\\(|\\)"</span>, <span class="st">""</span>, <span class="va">Glyco.features</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/separate.html" class="external-link">separate</a></span><span class="op">(</span><span class="va">Glyco.features</span>, into <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="cn">NA</span>, <span class="st">"Glycosylation.count"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">" "</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Entry</span>, <span class="va">Glycosylation.count</span><span class="op">)</span></span></code></pre></div>
<p>Add feature columns describing the glycoprotein and GO-RBP status of
the proteins.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">prot_ratios</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">prot_ratios</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Glycoprotein <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">prot_ratios</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">glycoproteins</span><span class="op">$</span><span class="va">Entry</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>GO.RBP <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">prot_ratios</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">rbps</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Glyco.RBP <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/interaction.html" class="external-link">interaction</a></span><span class="op">(</span><span class="va">Glycoprotein</span>, <span class="va">GO.RBP</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Glyco.RBP <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/recode.html" class="external-link">recode</a></span><span class="op">(</span></span>
<span>    <span class="va">Glyco.RBP</span>,</span>
<span>    <span class="st">'TRUE.TRUE'</span><span class="op">=</span><span class="st">'GO:RBGP'</span>,</span>
<span>    <span class="st">'FALSE.TRUE'</span><span class="op">=</span><span class="st">'GO:RBP'</span>,</span>
<span>    <span class="st">'TRUE.FALSE'</span><span class="op">=</span><span class="st">'Glycoprotein'</span>,</span>
<span>    <span class="st">'FALSE.FALSE'</span><span class="op">=</span><span class="st">'Other'</span><span class="op">)</span>,</span>
<span>    levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'GO:RBP'</span>, <span class="st">'GO:RBGP'</span>, <span class="st">'Other'</span>, <span class="st">'Glycoprotein'</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p>Below, we define a function to plot the ratios for each functional
sub-type of proteins.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot_ratios</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">obj</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">to_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-merge.html" class="external-link">merge</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">[</span>,<span class="st">'Glyco.RBP'</span>,drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>,</span>
<span>    by <span class="op">=</span> <span class="st">'row.names'</span></span>
<span>  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span>cols <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">Row.names</span>, <span class="va">Glyco.RBP</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">'sample'</span>, values_to <span class="op">=</span> <span class="st">'ratio'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span>, by.x <span class="op">=</span> <span class="st">'sample'</span>, by.y <span class="op">=</span> <span class="st">'row.names'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/is.finite.html" class="external-link">is.finite</a></span><span class="op">(</span><span class="va">ratio</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="va">p</span> <span class="op">&lt;-</span> <span class="va">to_plot</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Replicate</span>, y <span class="op">=</span> <span class="va">ratio</span>, </span>
<span>               group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/interaction.html" class="external-link">interaction</a></span><span class="op">(</span><span class="va">Glyco.RBP</span>, <span class="va">Replicate</span><span class="op">)</span>, </span>
<span>               colour <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">Glyco.RBP</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span>position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_dodge.html" class="external-link">position_dodge</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="../reference/theme_camprot.html">theme_camprot</a></span><span class="op">(</span>border <span class="op">=</span> <span class="cn">FALSE</span>, base_family <span class="op">=</span> <span class="st">'sans'</span>, base_size <span class="op">=</span> <span class="fl">15</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_colour_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="../reference/get_cat_palette.html">get_cat_palette</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span>, <span class="st">'black'</span><span class="op">)</span>, name <span class="op">=</span> <span class="st">''</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_hline</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0</span>, linetype <span class="op">=</span> <span class="fl">2</span>, colour <span class="op">=</span> <span class="st">'grey'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>      x <span class="op">=</span> <span class="st">"Replicate"</span>,</span>
<span>      y <span class="op">=</span> <span class="st">"RNase -/+ ratio"</span></span>
<span>    <span class="op">)</span></span>
<span>  </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="va">to_plot</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p>How do the protein ratios look pre-normalisation…</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_ratios</span><span class="op">(</span><span class="va">prot_ratios</span><span class="op">)</span></span></code></pre></div>
<p><img src="LFQ_parse_qc_protein_inference_files/figure-html/unnamed-chunk-32-1.png" width="768"></p>
<p>OK, so the glycoproteins are not centered at zero and there are
GO-annotated RBPs with negative log RNase -/+ ratios (as much as ~25% in
replicate 2)</p>
<p>Below, we perform the center median normalisation to the set of
reference proteins (here, the glycoproteins).</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">glycoprotein_medians</span> <span class="op">&lt;-</span> <span class="va">prot_ratios</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">prot_ratios</span><span class="op">)</span><span class="op">$</span><span class="va">Glyco.RBP</span> <span class="op">==</span> <span class="st">'Glycoprotein'</span>,<span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">camprotR</span><span class="fu">::</span><span class="fu"><a href="../reference/get_medians.html">get_medians</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">prot_ratios_norm</span> <span class="op">&lt;-</span> <span class="fu">camprotR</span><span class="fu">::</span><span class="fu"><a href="../reference/center_normalise_to_ref.html">center_normalise_to_ref</a></span><span class="op">(</span></span>
<span>  <span class="va">prot_ratios</span>,</span>
<span>  <span class="va">glycoprotein_medians</span>,</span>
<span>  center_to_zero <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># We want to center the glycoproteins around zero</span></span>
<span>  on_log_scale <span class="op">=</span> <span class="cn">TRUE</span> <span class="co"># The quantifications are on a log scale (log2 ratios)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>And plot the protein ratios post-normalisation.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_ratios</span><span class="op">(</span><span class="va">prot_ratios_norm</span><span class="op">)</span></span></code></pre></div>
<p><img src="LFQ_parse_qc_protein_inference_files/figure-html/unnamed-chunk-34-1.png" width="768"></p>
<p>Now, the median log2 RNase -/+ ratio for glycoproteins is zero for
all replicates and we have far fewer GO-annotated RBPs with negative log
RNase -/+ ratios.</p>
</div>
<div class="section level2 unnumbered">
<h2 class="unnumbered" id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<pre><code><span><span class="co">#&gt; R version 4.2.2 (2022-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.1 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] httr_1.4.4          tibble_3.1.8        dplyr_1.0.10       </span></span>
<span><span class="co">#&gt;  [4] tidyr_1.2.1         ggplot2_3.4.0       biobroom_1.30.0    </span></span>
<span><span class="co">#&gt;  [7] broom_1.0.2         gplots_3.1.3        naniar_0.6.1       </span></span>
<span><span class="co">#&gt; [10] MSnbase_2.24.2      ProtGenerics_1.30.0 mzR_2.32.0         </span></span>
<span><span class="co">#&gt; [13] Rcpp_1.0.9          Biobase_2.58.0      Biostrings_2.66.0  </span></span>
<span><span class="co">#&gt; [16] GenomeInfoDb_1.34.6 XVector_0.38.0      IRanges_2.32.0     </span></span>
<span><span class="co">#&gt; [19] S4Vectors_0.36.1    BiocGenerics_0.44.0 camprotR_0.0.0.9000</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] colorspace_2.0-3       ellipsis_0.3.2         visdat_0.5.3          </span></span>
<span><span class="co">#&gt;   [4] rprojroot_2.0.3        fs_1.5.2               clue_0.3-63           </span></span>
<span><span class="co">#&gt;   [7] farver_2.1.1           affyio_1.68.0          bit64_4.0.5           </span></span>
<span><span class="co">#&gt;  [10] AnnotationDbi_1.60.0   fansi_1.0.3            codetools_0.2-18      </span></span>
<span><span class="co">#&gt;  [13] ncdf4_1.21             doParallel_1.0.17      cachem_1.0.6          </span></span>
<span><span class="co">#&gt;  [16] impute_1.72.2          robustbase_0.95-0      knitr_1.41            </span></span>
<span><span class="co">#&gt;  [19] jsonlite_1.8.4         GO.db_3.16.0           cluster_2.1.4         </span></span>
<span><span class="co">#&gt;  [22] vsn_3.66.0             png_0.1-8              BiocManager_1.30.19   </span></span>
<span><span class="co">#&gt;  [25] compiler_4.2.2         backports_1.4.1        assertthat_0.2.1      </span></span>
<span><span class="co">#&gt;  [28] fastmap_1.1.0          limma_3.54.0           cli_3.5.0             </span></span>
<span><span class="co">#&gt;  [31] htmltools_0.5.4        tools_4.2.2            gtable_0.3.1          </span></span>
<span><span class="co">#&gt;  [34] glue_1.6.2             GenomeInfoDbData_1.2.9 affy_1.76.0           </span></span>
<span><span class="co">#&gt;  [37] rappdirs_0.3.3         MALDIquant_1.22        jquerylib_0.1.4       </span></span>
<span><span class="co">#&gt;  [40] pkgdown_2.0.7          vctrs_0.5.1            preprocessCore_1.60.1 </span></span>
<span><span class="co">#&gt;  [43] iterators_1.0.14       xfun_0.36              stringr_1.5.0         </span></span>
<span><span class="co">#&gt;  [46] lifecycle_1.0.3        iq_1.9.7               gtools_3.9.4          </span></span>
<span><span class="co">#&gt;  [49] XML_3.99-0.13          DEoptimR_1.0-11        zlibbioc_1.44.0       </span></span>
<span><span class="co">#&gt;  [52] MASS_7.3-58.1          scales_1.2.1           ragg_1.2.4            </span></span>
<span><span class="co">#&gt;  [55] pcaMethods_1.90.0      parallel_4.2.2         httr2_0.2.2           </span></span>
<span><span class="co">#&gt;  [58] yaml_2.3.6             curl_4.3.3             memoise_2.0.1         </span></span>
<span><span class="co">#&gt;  [61] gridExtra_2.3          UpSetR_1.4.0           sass_0.4.4            </span></span>
<span><span class="co">#&gt;  [64] RSQLite_2.2.20         stringi_1.7.8          highr_0.10            </span></span>
<span><span class="co">#&gt;  [67] desc_1.4.2             foreach_1.5.2          checkmate_2.1.0       </span></span>
<span><span class="co">#&gt;  [70] caTools_1.18.2         BiocParallel_1.32.5    rlang_1.0.6           </span></span>
<span><span class="co">#&gt;  [73] pkgconfig_2.0.3        systemfonts_1.0.4      bitops_1.0-7          </span></span>
<span><span class="co">#&gt;  [76] mzID_1.36.0            evaluate_0.19          lattice_0.20-45       </span></span>
<span><span class="co">#&gt;  [79] purrr_1.0.0            labeling_0.4.2         bit_4.0.5             </span></span>
<span><span class="co">#&gt;  [82] tidyselect_1.2.0       plyr_1.8.8             magrittr_2.0.3        </span></span>
<span><span class="co">#&gt;  [85] R6_2.5.1               generics_0.1.3         DBI_1.1.3             </span></span>
<span><span class="co">#&gt;  [88] pillar_1.8.1           withr_2.5.0            MsCoreUtils_1.10.0    </span></span>
<span><span class="co">#&gt;  [91] KEGGREST_1.38.0        RCurl_1.98-1.9         crayon_1.5.2          </span></span>
<span><span class="co">#&gt;  [94] KernSmooth_2.23-20     utf8_1.2.2             rmarkdown_2.19        </span></span>
<span><span class="co">#&gt;  [97] grid_4.2.2             blob_1.2.3             uniprotREST_0.0.0.9000</span></span>
<span><span class="co">#&gt; [100] digest_0.6.31          textshaping_0.3.6      munsell_0.5.0         </span></span>
<span><span class="co">#&gt; [103] bslib_0.4.2</span></span></code></pre>
</div>
<div class="section level2 unnumbered">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-http://zotero.org/users/5634351/items/87YIDK69" class="csl-entry">
Cox, Jürgen, Marco Y. Hein, Christian A. Luber, Igor Paron, Nagarjuna
Nagaraj, and Matthias Mann. 2014. <span>“Accurate Proteome-Wide
Label-Free Quantification by Delayed Normalization and Maximal Peptide
Ratio Extraction, Termed MaxLFQ*.”</span> <em>Molecular &amp; Cellular
Proteomics</em> 13 (9): 2513–26. <a href="https://doi.org/10.1074/mcp.M113.031591" class="external-link">https://doi.org/10.1074/mcp.M113.031591</a>.
</div>
<div id="ref-http://zotero.org/users/5634351/items/EJHWF46N" class="csl-entry">
Queiroz, Rayner M. L., Tom Smith, Eneko Villanueva, Maria Marti-Solano,
Mie Monti, Mariavittoria Pizzinga, Dan-Mircea Mirea, et al. 2019.
<span>“Comprehensive Identification of RNA–Protein Interactions in Any
Organism Using Orthogonal Organic Phase Separation (OOPS).”</span>
<em>Nature Biotechnology</em> 37 (2): 169. <a href="https://doi.org/10.1038/s41587-018-0001-2" class="external-link">https://doi.org/10.1038/s41587-018-0001-2</a>.
</div>
<div id="ref-http://zotero.org/users/5634351/items/FZN3QTTZ" class="csl-entry">
Sticker, Adriaan, Ludger Goeminne, Lennart Martens, and Lieven Clement.
2020. <span>“Robust Summarization and Inference in Proteome-wide
Label-free Quantification.”</span> <em>Molecular &amp; cellular
proteomics: MCP</em> 19 (7): 1209–19. <a href="https://doi.org/10.1074/mcp.RA119.001624" class="external-link">https://doi.org/10.1074/mcp.RA119.001624</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Charlotte Dawson, Tom Smith.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
