<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Processing and QC of SILAC data • camprotR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Processing and QC of SILAC data">
<meta property="og:description" content="camprotR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">camprotR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/LFQ_parse_qc_protein_inference.html">LFQ QC and summarisation to protein-level abundance</a>
    </li>
    <li>
      <a href="../articles/SILAC.html">Processing and QC of SILAC data</a>
    </li>
    <li>
      <a href="../articles/TMT_PSM_QC_Summarisation.html">TMT QC PSM-level quantification and summarisation to protein-level abundance</a>
    </li>
    <li>
      <a href="../articles/crap.html">cRAP databases</a>
    </li>
    <li>
      <a href="../articles/msnset.html">Working with MSnSets</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Processing and QC of SILAC data</h1>
                        <h4 data-toc-skip class="author">Tom Smith</h4>
            
            <h4 data-toc-skip class="date">2022-04-19</h4>
      
      
      <div class="hidden name"><code>SILAC.Rmd</code></div>

    </div>

    
    
<div class="section level3">
<h3 id="load-dependencies">Load dependencies<a class="anchor" aria-label="anchor" href="#load-dependencies"></a>
</h3>
<p>Load the required libraries.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biostrings" class="external-link">Biostrings</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">camprotR</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://lgatto.github.io/MSnbase" class="external-link">MSnbase</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">DEqMS</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/limma" class="external-link">limma</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="preamble">Preamble<a class="anchor" aria-label="anchor" href="#preamble"></a>
</h2>
<p>Stable Isotope Labelling by/with Amino acids in Cell culture (SILAC) is a form of quantitative proteomics where different conditions are quantified in the same run by differential metabolic labelling of proteins using amino acids containing stable isotopes <span class="citation">(Ong et al. 2002)</span>. A typical SILAC experiment involved growing cells in two different types of SILAC media, one containing ‘light’ arginine and lysine, and the other containing ‘heavy’ arginine and lysine. Over time the cells incorporate these light or heavy amino acids into their proteins which gives resulting downstream peptides different masses which can be detected by LC-MS.Initially designed to provide pairwise comparisons between cell cultures, it has now been extended to more than two labels and even whole organisms <span class="citation">(Krüger et al. 2008)</span>.</p>
<p>This elegant experimental design enables quantification of peptide/protein abundance ratios between conditions with very little technical variation, since the samples from separate conditions are pooled as early as possible in the sample handling process. Furthermore, the use of different isotope labels has been extended to study the turnover of protein (e.g switch from one label to another) in pulsed SILAC, and relative turnover between conditions (e.g two condition on the same label and then switch each condition to a different label; requires triple SILAC).</p>
<p>The analysis of SILAC data is relatively straightforward, since technical noise is low and normalisation is not normally required. Typically, one is interested in the intensity (peak area or peak height depending on your PD settings) ratio of the light (L) and heavy (H) MS peaks for a given peptide. This ratio forms the quantification value which we wish to perform statistical tests and exploratory analysis on.</p>
</div>
<div class="section level2">
<h2 id="asssessing-the-silac-isotope-incorporation-rate">Asssessing the SILAC isotope incorporation rate<a class="anchor" aria-label="anchor" href="#asssessing-the-silac-isotope-incorporation-rate"></a>
</h2>
<p>Before performing a SILAC experiment, it is necessary to confirm complete (e.g &gt; 95%) incorporation of the supplemented isotopically labelled amino acids into the cellular proteome.</p>
<p>We do this with <code><a href="../reference/estimate_incorporation.html">camprotR::estimate_incorporation()</a></code> below. Calculating the incorporation for a given peptide is simple, e.g H/(H+L) for a typical two label experiment. However, for very highly incorporated samples, the true intensity of the L peptide ions will be close to zero, which can lead to erroneous assignment of other ions to the L peptide. This function supports the calculation of incorporation from defined mixes of heavy and light labelled material, which makes the estimation of incorporation rate more robust since both isotopes should exist for all peptides.</p>
<p>Here, heavy and light labelled cells were mixed 1:1.</p>
<p>Importantly, note that there is a good correlation between the light and heavy intensities, regardless whether either or both isotopes were from a spectrum match (the other option being a mass shift relative to the peptide-spectrum match (PSM) for the cognate isotope). This indicates that the quantification values are from true isotope pairs.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">crap_fasta_inf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>
  <span class="st">"extdata"</span>, <span class="st">"cRAP_20190401.fasta.gz"</span>, 
  package <span class="op">=</span> <span class="st">"camprotR"</span>
<span class="op">)</span>

<span class="va">incorporation_results</span> <span class="op">&lt;-</span> <span class="fu">camprotR</span><span class="fu">::</span><span class="fu"><a href="../reference/estimate_incorporation.html">estimate_incorporation</a></span><span class="op">(</span>
  <span class="co"># can replace with PD output using read.delim(PSMs.txt)</span>
  psm_input <span class="op">=</span> <span class="va">psm_silac_p4</span>, 
  <span class="co"># can replace with PD output using read.delim(PeptideGroups.txt)</span>
  peptide_input <span class="op">=</span> <span class="va">pep_silac_p4</span>, 
  crap_fasta <span class="op">=</span> <span class="va">crap_fasta_inf</span>,
  mix <span class="op">=</span> <span class="fl">1</span> <span class="co"># This incorporation test was performed with a 1:1 H/L mix</span>
<span class="op">)</span>
<span class="co">#&gt; camprotR::silac_psm_seq_int output has changed.</span>
<span class="co">#&gt;   Columns indicating whether quantification is from PSM are now prefixed with</span>
<span class="co">#&gt;   "matched", not "Sequenced", and tallys of PSMs per isotope are included.</span>
<span class="co">#&gt;   Interference is not included by default (set include_interference=TRUE)</span>
<span class="co">#&gt; Parsing features...</span>
<span class="co">#&gt; 470 features found from 100 master proteins =&gt; Input</span>
<span class="co">#&gt; 242 cRAP proteins supplied</span>
<span class="co">#&gt; 0 proteins identified as 'cRAP associated'</span>
<span class="co">#&gt; 470 features found from 100 master proteins =&gt; cRAP features removed</span>
<span class="co">#&gt; 306 features found from 93 master proteins =&gt; features without quantification removed</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">incorporation_results</span><span class="op">$</span><span class="va">HL_correlation</span><span class="op">)</span></code></pre></div>
<p><img src="SILAC_files/figure-html/unnamed-chunk-3-1.png" width="576"></p>
<p>The median observed incorporation for the mixed sample is 50.8 at the peptide-level and 50.5 at the protein level. Correcting for the mixing proportions this gives us incorporation rate estimates just over 100%, which suggests the true incorporation rate is likely to be very close to 100%.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">incorporation_results</span><span class="op">$</span><span class="va">peptide_incorporation</span><span class="op">)</span>
<span class="co">#&gt; `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre></div>
<p><img src="SILAC_files/figure-html/unnamed-chunk-4-1.png" width="384"></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">incorporation_results</span><span class="op">$</span><span class="va">protein_incorporation</span><span class="op">)</span>
<span class="co">#&gt; `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre></div>
<p><img src="SILAC_files/figure-html/unnamed-chunk-4-2.png" width="384"></p>
</div>
<div class="section level2">
<h2 id="processing-and-qc-of-silac-experimental-data">Processing and QC of SILAC experimental data<a class="anchor" aria-label="anchor" href="#processing-and-qc-of-silac-experimental-data"></a>
</h2>
<p>Now, onto some experimental data. Here, we process data from an OOPS <span class="citation">(Queiroz et al. 2019)</span> experiment designed to identify proteins that are significantly enriched upon UV crosslinking (CL) vs non-crosslinked control cells (NC). RNA-binding proteins (RBPs) should be much more abundant in CL samples, since this covalently crosslinks the RBPs to RNA, and retains them in the OOPS interface, from which the sample is taken. In total, we have 4 SILAC runs, representing 4 separate plates of U-2 OS cells. Two plates (replicate 1 &amp; 2) were Heavy for CL and two plates (replicates 3 &amp; 4) were Light for CL. This is called a ‘label’ swap/switch and helps to ensure the results are independent of the labelling scheme.</p>
<p>The first step is to read in the data and remove contaminant proteins and features without quantification data. These were defined using the cRAP database. Below, we parse the cRAP FASTA to extract the IDs for the cRAP proteins, in both ‘cRAP’ format and Uniprot IDs for these proteins.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Load the cRAP FASTA used for the PD search</span>
<span class="va">crap_fasta</span> <span class="op">&lt;-</span> <span class="fu">Biostrings</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html" class="external-link">fasta.index</a></span><span class="op">(</span><span class="va">crap_fasta_inf</span>, seqtype <span class="op">=</span> <span class="st">"AA"</span><span class="op">)</span>

<span class="co"># Define a base R version of stringr::str_extract_all()</span>
<span class="co"># to avoid loading the whole stringr package</span>
<span class="va">str_extract_all</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">pattern</span>, <span class="va">string</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gregexpr</a></span><span class="op">(</span><span class="va">pattern</span>, <span class="va">string</span>, perl <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
    <span class="fu"><a href="https://rdrr.io/r/base/regmatches.html" class="external-link">regmatches</a></span><span class="op">(</span><span class="va">string</span>, <span class="va">.</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="op">)</span>
<span class="op">}</span>

<span class="co"># Extract the non-cRAP UniProt accessions associated with each cRAP protein</span>
<span class="va">crap_accessions</span> <span class="op">&lt;-</span> <span class="va">crap_fasta</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">desc</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu">str_extract_all</span><span class="op">(</span><span class="st">"(?&lt;=\\|).*?(?=\\|)"</span>, <span class="va">.</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>We can then supply these cRAP protein IDs to <code><a href="../reference/parse_features.html">camprotR::parse_features</a></code> which will remove features which may originate from contaminants, as well as features which don’t have a unique master protein. See <code><a href="../reference/parse_features.html">?parse_features</a></code> for further details, including the removal of ‘associated cRAP.’ <code>parse_features</code> will output messages about the number of features at each stage in the filtering.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># These files are part of the camprotR package</span>
<span class="va">pep_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">pep_oops_1</span>, <span class="va">pep_oops_2</span>, <span class="va">pep_oops_3</span>, <span class="va">pep_oops_4</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pep_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'pep_1'</span>, <span class="st">'pep_2'</span>, <span class="st">'pep_3'</span>, <span class="st">'pep_4'</span><span class="op">)</span>

<span class="va">pep_data_parsed</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span>
  <span class="va">pep_data</span>, <span class="kw">function</span><span class="op">(</span><span class="va">infdata</span><span class="op">)</span> <span class="op">{</span>
    <span class="fu"><a href="../reference/parse_features.html">parse_features</a></span><span class="op">(</span><span class="va">infdata</span>,
                   silac <span class="op">=</span> <span class="cn">TRUE</span>,
                   level <span class="op">=</span> <span class="st">'peptide'</span>,
                   crap_proteins <span class="op">=</span> <span class="va">crap_accessions</span>,
                   unique_master <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
  <span class="op">}</span>
<span class="op">)</span>
<span class="co">#&gt; Parsing features...</span>
<span class="co">#&gt; 550 features found from 101 master proteins =&gt; Input</span>
<span class="co">#&gt; 242 cRAP proteins supplied</span>
<span class="co">#&gt; 6 proteins identified as 'cRAP associated'</span>
<span class="co">#&gt; 526 features found from 99 master proteins =&gt; cRAP features removed</span>
<span class="co">#&gt; 525 features found from 98 master proteins =&gt; associated cRAP features removed</span>
<span class="co">#&gt; 355 features found from 91 master proteins =&gt; features without quantification removed</span>
<span class="co">#&gt; Parsing features...</span>
<span class="co">#&gt; 547 features found from 101 master proteins =&gt; Input</span>
<span class="co">#&gt; 242 cRAP proteins supplied</span>
<span class="co">#&gt; 8 proteins identified as 'cRAP associated'</span>
<span class="co">#&gt; 527 features found from 99 master proteins =&gt; cRAP features removed</span>
<span class="co">#&gt; 525 features found from 98 master proteins =&gt; associated cRAP features removed</span>
<span class="co">#&gt; 379 features found from 92 master proteins =&gt; features without quantification removed</span>
<span class="co">#&gt; Parsing features...</span>
<span class="co">#&gt; 580 features found from 101 master proteins =&gt; Input</span>
<span class="co">#&gt; 242 cRAP proteins supplied</span>
<span class="co">#&gt; 9 proteins identified as 'cRAP associated'</span>
<span class="co">#&gt; 558 features found from 98 master proteins =&gt; cRAP features removed</span>
<span class="co">#&gt; 558 features found from 98 master proteins =&gt; associated cRAP features removed</span>
<span class="co">#&gt; 397 features found from 94 master proteins =&gt; features without quantification removed</span>
<span class="co">#&gt; Parsing features...</span>
<span class="co">#&gt; 683 features found from 102 master proteins =&gt; Input</span>
<span class="co">#&gt; 242 cRAP proteins supplied</span>
<span class="co">#&gt; 9 proteins identified as 'cRAP associated'</span>
<span class="co">#&gt; 657 features found from 99 master proteins =&gt; cRAP features removed</span>
<span class="co">#&gt; 655 features found from 98 master proteins =&gt; associated cRAP features removed</span>
<span class="co">#&gt; 502 features found from 98 master proteins =&gt; features without quantification removed</span></code></pre></div>
<p>Next, we add new columns to describe the intensities with respect to the CL and NC conditions, based on how H &amp; L map to CL &amp; NC for each replicate.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pep_data_parsed</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pep_data_parsed</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">experiment</span><span class="op">)</span> <span class="op">{</span>
  <span class="co"># In this case, the replicate number is after the '_' so we extract like so</span>
  <span class="va">rep_n</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="va">experiment</span>, split<span class="op">=</span><span class="st">'_'</span><span class="op">)</span>, <span class="st">'[['</span>, <span class="fl">2</span><span class="op">)</span> 
  
  <span class="va">.data</span> <span class="op">&lt;-</span> <span class="va">pep_data_parsed</span><span class="op">[[</span><span class="va">experiment</span><span class="op">]</span><span class="op">]</span>
  
  <span class="co"># Identify the column names for the light and heavy intensities</span>
  <span class="co"># (these names are not consistent between samples, yours may be different!)</span>
  <span class="va">abundance_light_col</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">'Abundance.*.Light.Sample'</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">.data</span><span class="op">)</span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
  <span class="va">abundance_heavy_col</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">'Abundance.*.Heavy.Sample'</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">.data</span><span class="op">)</span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
  
  <span class="co"># Label-swap info in cell above defines whether heavy or light is CL</span>
  <span class="va">cl_col</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">rep_n</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, <span class="va">abundance_heavy_col</span>, <span class="va">abundance_light_col</span><span class="op">)</span> 
  <span class="va">nc_col</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">rep_n</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, <span class="va">abundance_light_col</span>, <span class="va">abundance_heavy_col</span><span class="op">)</span>
  
  <span class="va">.data</span> <span class="op">&lt;-</span> <span class="va">.data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="co"># Add new columns with CL or NC intensities </span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span><span class="st">'CL'</span> <span class="op">=</span> <span class="op">!</span><span class="op">!</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/tidyeval.html" class="external-link">sym</a></span><span class="op">(</span><span class="va">cl_col</span><span class="op">)</span>,
           <span class="st">'NC'</span> <span class="op">=</span> <span class="op">!</span><span class="op">!</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/tidyeval.html" class="external-link">sym</a></span><span class="op">(</span><span class="va">nc_col</span><span class="op">)</span>,
           replicate <span class="op">=</span> <span class="va">rep_n</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="co"># And subset to these columns</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Master.Protein.Accessions</span>, <span class="va">CL</span>, <span class="va">NC</span>, <span class="va">replicate</span>, <span class="va">Sequence</span>, <span class="va">Modifications</span><span class="op">)</span>
  
  <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">.data</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>

<span class="co"># Re-annotate with names</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pep_data_parsed</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pep_data</span><span class="op">)</span></code></pre></div>
<p>Having added the replicate number as a column previously, we bind together the rows from all replicates and calculate the CL/NC ratio using <code><a href="../reference/get_ratio.html">camprotR::get_ratio</a></code>, which will also add a <code>missing</code> column to describe whether one of the quantification values is missing.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">abundance_data_complete</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="st">'rbind'</span>, <span class="va">pep_data_parsed</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">remove_rownames</a></span><span class="op">(</span><span class="op">)</span>

<span class="va">ratios</span> <span class="op">&lt;-</span> <span class="va">abundance_data_complete</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/is.finite.html" class="external-link">is.finite</a></span><span class="op">(</span><span class="va">CL</span><span class="op">)</span> <span class="op">|</span> <span class="fu"><a href="https://rdrr.io/r/base/is.finite.html" class="external-link">is.finite</a></span><span class="op">(</span><span class="va">NC</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>CL <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">CL</span><span class="op">)</span>, NC <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">NC</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="../reference/get_ratio.html">get_ratio</a></span><span class="op">(</span><span class="va">CL</span>, <span class="va">NC</span>, bind <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> </code></pre></div>
<p>Below, we tally and plot the missing values per replicate for each method. Note that ~40% of peptides have a missing quantification value and it’s much more likely to be missing in NC than CL. This is what we expect since the OOPS method is designed to capture RBPs that are UV crosslinked to RNA, so without crosslinking, there should be very little protein present in the interface.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ratios</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">replicate</span>, <span class="va">missing</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">tally</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">replicate</span>, y <span class="op">=</span> <span class="va">n</span>, fill <span class="op">=</span> <span class="va">missing</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">'identity'</span>, position <span class="op">=</span> <span class="st">'fill'</span>, colour <span class="op">=</span> <span class="st">'grey50'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/theme_camprot.html">theme_camprot</a></span><span class="op">(</span>border <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">'Replicate'</span>, y <span class="op">=</span> <span class="st">'Fraction'</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="../reference/get_cat_palette.html">get_cat_palette</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span>, name <span class="op">=</span> <span class="st">''</span><span class="op">)</span></code></pre></div>
<p><img src="SILAC_files/figure-html/unnamed-chunk-9-1.png" width="576"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">ratios</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">replicate</span>, <span class="va">missing</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">tally</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">replicate</span>, y <span class="op">=</span> <span class="va">n</span>, fill <span class="op">=</span> <span class="va">missing</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">'identity'</span>, position <span class="op">=</span> <span class="st">'stack'</span>, colour <span class="op">=</span> <span class="st">'grey50'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/theme_camprot.html">theme_camprot</a></span><span class="op">(</span>border <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">'Replicate'</span>, y <span class="op">=</span> <span class="st">'Peptides'</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="../reference/get_cat_palette.html">get_cat_palette</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span>, name <span class="op">=</span> <span class="st">''</span><span class="op">)</span></code></pre></div>
<p><img src="SILAC_files/figure-html/unnamed-chunk-9-2.png" width="576"></p>
<p>Next, we plot the distribution for the ratios. Note that there are at least two populations, with the clearest two being:</p>
<ol style="list-style-type: decimal">
<li>No difference between CL and NC</li>
<li>Higher abundance in CL, with the modal CL:NC ratio being around 6-fold</li>
</ol>
<p>We expect that population 1 are the proteins which are not RBPs and population 2 are the true RBPs.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ratios</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">ratio</span>, colour <span class="op">=</span> <span class="va">replicate</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, linetype <span class="op">=</span> <span class="fl">2</span>, colour <span class="op">=</span> <span class="st">'grey50'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_density.html" class="external-link">geom_density</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/theme_camprot.html">theme_camprot</a></span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">15</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Ratio CL:NC (Log2)"</span>, y <span class="op">=</span> <span class="st">"Density"</span>, colour <span class="op">=</span> <span class="st">"Replicate"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_colour_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="../reference/get_cat_palette.html">get_cat_palette</a></span><span class="op">(</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Warning: Removed 638 rows containing non-finite values (stat_density).</span></code></pre></div>
<p><img src="SILAC_files/figure-html/unnamed-chunk-10-1.png" width="576"></p>
<p>We can also consider the relationship between the CL and NC intensities. Again, we see these two populations, with the first being on the dashed line (equality).</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ratios</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">CL</span>, y <span class="op">=</span> <span class="va">NC</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>colour <span class="op">=</span> <span class="va">replicate</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_abline</a></span><span class="op">(</span>slope <span class="op">=</span> <span class="fl">1</span>, linetype <span class="op">=</span> <span class="fl">2</span>, colour <span class="op">=</span> <span class="st">'grey50'</span><span class="op">)</span> <span class="op">+</span> <span class="co"># line at CL==NC</span>
  <span class="fu"><a href="../reference/theme_camprot.html">theme_camprot</a></span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">15</span>, border <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_colour_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="../reference/get_cat_palette.html">get_cat_palette</a></span><span class="op">(</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Warning: Removed 638 rows containing missing values (geom_point).</span></code></pre></div>
<p><img src="SILAC_files/figure-html/unnamed-chunk-11-1.png" width="576"></p>
<p>Given the large number of missing values, there is a concern that protein abundance in NC may be very low and, therefore, that quantification values from NC peptides identified by mass shift rather than peptide spectrum matching could be falsely assigning ions to peptides. Note that peptide sequences identified by PSM are referred to as having been ‘sequenced’ in some code blocks below.</p>
<p>To get the information about what quantification values come from PSMs, we need to interrogate the PSM-level output.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># These files are part of the camprotR package</span>
<span class="va">psm_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">psm_oops_1</span>, <span class="va">psm_oops_2</span>, <span class="va">psm_oops_3</span>, <span class="va">psm_oops_4</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">psm_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'psm_1'</span>, <span class="st">'psm_2'</span>, <span class="st">'psm_3'</span>, <span class="st">'psm_4'</span><span class="op">)</span></code></pre></div>
<p>Below we use <code><a href="../reference/silac_psm_seq_int.html">camprotR::silac_psm_seq_int</a></code> to identify which quantification values are from PSMs. This function also extracts the interference/co-isolation value for the PSMs, which is not available through the peptide-level output, though we won’t use that information here.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">psm_matched_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span>
  <span class="va">psm_data</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
    <span class="co"># If you export from PD including the 'Sequence' column, </span>
    <span class="co"># you don't need this step.</span>
    <span class="co"># Below, we convert the Annotated.Sequence column into a Sequence column</span>
    <span class="va">x</span><span class="op">$</span><span class="va">Sequence</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">Annotated.Sequence</span>, split <span class="op">=</span> <span class="st">'\\.'</span><span class="op">)</span>, <span class="st">'[['</span>, <span class="fl">1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
      <span class="fu"><a href="https://rdrr.io/r/base/chartr.html" class="external-link">toupper</a></span><span class="op">(</span><span class="op">)</span>
    
    <span class="fu">camprotR</span><span class="fu">::</span><span class="fu"><a href="../reference/silac_psm_seq_int.html">silac_psm_seq_int</a></span><span class="op">(</span><span class="va">x</span>, sequence_col <span class="op">=</span> <span class="st">'Sequence'</span><span class="op">)</span>
  <span class="op">}</span>
<span class="op">)</span>
<span class="co">#&gt; camprotR::silac_psm_seq_int output has changed.</span>
<span class="co">#&gt;   Columns indicating whether quantification is from PSM are now prefixed with</span>
<span class="co">#&gt;   "matched", not "Sequenced", and tallys of PSMs per isotope are included.</span>
<span class="co">#&gt;   Interference is not included by default (set include_interference=TRUE)</span>
<span class="co">#&gt; camprotR::silac_psm_seq_int output has changed.</span>
<span class="co">#&gt;   Columns indicating whether quantification is from PSM are now prefixed with</span>
<span class="co">#&gt;   "matched", not "Sequenced", and tallys of PSMs per isotope are included.</span>
<span class="co">#&gt;   Interference is not included by default (set include_interference=TRUE)</span>
<span class="co">#&gt; camprotR::silac_psm_seq_int output has changed.</span>
<span class="co">#&gt;   Columns indicating whether quantification is from PSM are now prefixed with</span>
<span class="co">#&gt;   "matched", not "Sequenced", and tallys of PSMs per isotope are included.</span>
<span class="co">#&gt;   Interference is not included by default (set include_interference=TRUE)</span>
<span class="co">#&gt; camprotR::silac_psm_seq_int output has changed.</span>
<span class="co">#&gt;   Columns indicating whether quantification is from PSM are now prefixed with</span>
<span class="co">#&gt;   "matched", not "Sequenced", and tallys of PSMs per isotope are included.</span>
<span class="co">#&gt;   Interference is not included by default (set include_interference=TRUE)</span></code></pre></div>
<p>Then we bind together the spectrum matched data into one data frame, ready to merge with the peptide quantification.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">all_psm_matched_data</span> <span class="op">&lt;-</span> <span class="va">psm_matched_data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">psm_name</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">rep_n</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="va">psm_name</span>, split<span class="op">=</span><span class="st">'_'</span><span class="op">)</span>, <span class="st">'[['</span>, <span class="fl">2</span><span class="op">)</span> 
    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">rep_n</span><span class="op">)</span>
    <span class="va">psm_matched_data</span><span class="op">[[</span><span class="va">psm_name</span><span class="op">]</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>replicate <span class="op">=</span> <span class="va">rep_n</span><span class="op">)</span>
  <span class="op">}</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html" class="external-link">bind_rows</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; [1] "1"</span>
<span class="co">#&gt; [1] "2"</span>
<span class="co">#&gt; [1] "3"</span>
<span class="co">#&gt; [1] "4"</span></code></pre></div>
<p>Below, we merge the quantification and PSM information and make new columns to describe whether the CL/NC quantification is from a PSM.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># merge the matched information and add new columns for the CL/NC sequenced information</span>
<span class="va">ratios_matched</span> <span class="op">&lt;-</span> <span class="va">ratios</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="co"># Update the modifications column for the peptide object so it </span>
  <span class="co"># doesn't include SILAC modifications </span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Modifications <span class="op">=</span> <span class="fu"><a href="../reference/remove_silac_modifications.html">remove_silac_modifications</a></span><span class="op">(</span><span class="va">Modifications</span>, level <span class="op">=</span> <span class="st">'peptide'</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="co"># Update the sequence column to all uppercase</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rowwise.html" class="external-link">rowwise</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Sequence <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/chartr.html" class="external-link">toupper</a></span><span class="op">(</span><span class="va">Sequence</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="co"># Merge with the sequenced information</span>
  <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">all_psm_matched_data</span>, 
        by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Sequence'</span>, <span class="st">'Modifications'</span>, <span class="st">'replicate'</span><span class="op">)</span><span class="op">)</span>  <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="co"># Add new columns with CL/NC and CL/NC sequenced information</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span><span class="st">'Matched_CL'</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">replicate</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>,
                                 <span class="va">matched_Heavy</span>, <span class="va">matched_Light</span><span class="op">)</span>,
         <span class="st">'Matched_NC'</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">replicate</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>,
                                 <span class="va">matched_Light</span>, <span class="va">matched_Heavy</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="co"># And subset to these columns</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Master.Protein.Accessions</span>,
         <span class="va">Sequence</span>, 
         <span class="va">Modifications</span>,
         <span class="va">replicate</span>,
         <span class="va">ratio</span>,
         <span class="va">CL</span>, 
         <span class="va">NC</span>,
         <span class="va">Matched_CL</span>,
         <span class="va">Matched_NC</span>,
         <span class="va">missing</span><span class="op">)</span>

<span class="co"># Add a single column to describe the sequenced information</span>
<span class="va">ratios_matched</span> <span class="op">&lt;-</span> <span class="va">ratios_matched</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>
    matched <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/interaction.html" class="external-link">interaction</a></span><span class="op">(</span><span class="va">Matched_CL</span>, <span class="va">Matched_NC</span><span class="op">)</span>,
    matched <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/recode.html" class="external-link">recode</a></span><span class="op">(</span><span class="va">matched</span>,
                            <span class="st">'TRUE.TRUE'</span><span class="op">=</span><span class="st">'Both spectrum matched'</span>,
                            <span class="st">'TRUE.FALSE'</span><span class="op">=</span><span class="st">'CL spectrum matched'</span>,
                            <span class="st">'FALSE.TRUE'</span><span class="op">=</span><span class="st">'NC spectrum matched'</span><span class="op">)</span>, 
                     levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Both spectrum matched'</span>, 
                                <span class="st">'CL spectrum matched'</span>, 
                                <span class="st">'NC spectrum matched'</span><span class="op">)</span><span class="op">)</span>
  <span class="op">)</span></code></pre></div>
<p>Below, we consider how often the ratio comes from a peptide where both CL and NC were sequenced, or just one sequenced, and the other one therefore being by mass shift. Note that a minority of peptides have both CL and NC sequenced and the majority are just sequenced in CL.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="va">ratios_matched</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/is.finite.html" class="external-link">is.finite</a></span><span class="op">(</span><span class="va">ratio</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">replicate</span>, <span class="va">matched</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">tally</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">replicate</span>, y <span class="op">=</span> <span class="va">n</span>, fill <span class="op">=</span> <span class="va">matched</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/theme_camprot.html">theme_camprot</a></span><span class="op">(</span>border <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="../reference/get_cat_palette.html">get_cat_palette</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span>, name <span class="op">=</span> <span class="st">''</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">'Replicate'</span><span class="op">)</span>

<span class="va">p1</span> <span class="op">&lt;-</span> <span class="va">p</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">'identity'</span>, colour <span class="op">=</span> <span class="st">'grey50'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">'Peptides'</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p1</span><span class="op">)</span></code></pre></div>
<p><img src="SILAC_files/figure-html/unnamed-chunk-16-1.png" width="576"></p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="va">p</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">'identity'</span>, position <span class="op">=</span> <span class="st">'fill'</span>, colour <span class="op">=</span> <span class="st">'grey50'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">'Fraction of peptides'</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p2</span><span class="op">)</span></code></pre></div>
<p><img src="SILAC_files/figure-html/unnamed-chunk-16-2.png" width="576"></p>
<p>We can then see consider how the CL vs NC correlation relates to the ‘sequenced’ information. Note that the correlation is much poorer when just CL is sequenced. This is a concern, since one plausible explanation would be that many of the NC quantification values obtained when CL is identified by spectrum matching, but NC is identified by mass shift are from ions erroneously assigned to a peptide.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ratios_matched</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/is.finite.html" class="external-link">is.finite</a></span><span class="op">(</span><span class="va">CL</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/is.finite.html" class="external-link">is.finite</a></span><span class="op">(</span><span class="va">NC</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">matched</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>cor <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cor</a></span><span class="op">(</span><span class="va">CL</span>, <span class="va">NC</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 2</span></span>
<span class="co">#&gt;   matched                 cor</span>
<span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Both spectrum matched 0.554</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">2</span> CL spectrum matched   0.409</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">3</span> NC spectrum matched   0.613</span>

<span class="va">ratios_matched</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">CL</span>, y <span class="op">=</span> <span class="va">NC</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>colour <span class="op">=</span> <span class="va">replicate</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_abline</a></span><span class="op">(</span>slope <span class="op">=</span> <span class="fl">1</span>, linetype <span class="op">=</span> <span class="fl">2</span>, colour <span class="op">=</span> <span class="st">'grey50'</span><span class="op">)</span> <span class="op">+</span> <span class="co"># line at CL==NC</span>
  <span class="fu"><a href="../reference/theme_camprot.html">theme_camprot</a></span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">15</span>, border <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span> <span class="va">matched</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_continuous</a></span><span class="op">(</span>breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">14</span>, <span class="fl">26</span>, <span class="fl">4</span><span class="op">)</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">14</span>, <span class="fl">26</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_continuous</a></span><span class="op">(</span>breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">14</span>, <span class="fl">26</span>, <span class="fl">4</span><span class="op">)</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">14</span>, <span class="fl">26</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_colour_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="../reference/get_cat_palette.html">get_cat_palette</a></span><span class="op">(</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Warning: Removed 628 rows containing missing values (geom_point).</span></code></pre></div>
<p><img src="SILAC_files/figure-html/unnamed-chunk-17-1.png" width="768"></p>
<p>Whenever a SILAC experiment is performed where complete absence of a protein in one condition is likely, it is recommended to perform this QC step. If the correlation is very poor when just one condition is spectrum matched, it may be necessary to exclude these peptides.</p>
<p>In this vignette example, we will proceed with all the peptides, but retain information about the spectrum matching in the protein quantification, should we need to investigate this further.</p>
<p>Next, we make an <code>MSnSet</code> from the peptide-level CL/NC ratios.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Remove peptides without both CL + NC quantified</span>
<span class="va">flt_ratios</span> <span class="op">&lt;-</span> <span class="va">ratios_matched</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/is.finite.html" class="external-link">is.finite</a></span><span class="op">(</span><span class="va">ratio</span><span class="op">)</span><span class="op">)</span>

<span class="co"># Create a wide table with unique ids as row names</span>
<span class="va">flt_ratios_wide</span> <span class="op">&lt;-</span> <span class="va">flt_ratios</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Master.Protein.Accessions</span>, <span class="va">Sequence</span>, <span class="va">Modifications</span>,
         <span class="va">replicate</span>, <span class="va">Matched_NC</span>, <span class="va">ratio</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_wider.html" class="external-link">pivot_wider</a></span><span class="op">(</span>names_from <span class="op">=</span> <span class="st">"replicate"</span>, values_from <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Matched_NC"</span>, <span class="st">"ratio"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/unite.html" class="external-link">unite</a></span><span class="op">(</span><span class="va">id</span>, <span class="va">Sequence</span>, <span class="va">Modifications</span>, remove <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"id"</span><span class="op">)</span>

<span class="co"># Create expression matrix (exprs)</span>
<span class="va">exprs_data</span> <span class="op">&lt;-</span> <span class="va">flt_ratios_wide</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">matches</a></span><span class="op">(</span><span class="st">"ratio_[1-4]"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="co"># select unique ids and ratio data</span>
  <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="op">)</span>

<span class="co"># Create feature metadata data frame (fData)</span>
<span class="va">feat_data</span> <span class="op">&lt;-</span> <span class="va">flt_ratios_wide</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">matches</a></span><span class="op">(</span><span class="st">"ratio"</span><span class="op">)</span><span class="op">)</span> <span class="co"># select unique ids and everything but ratio data</span>

<span class="co"># Create MSnSet</span>
<span class="va">pep_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/MSnSet-class.html" class="external-link">MSnSet</a></span><span class="op">(</span>exprs <span class="op">=</span> <span class="va">exprs_data</span>, fData <span class="op">=</span> <span class="va">feat_data</span><span class="op">)</span>

<span class="co"># Add replicate number to phenotype data</span>
<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">pep_res</span><span class="op">)</span><span class="op">$</span><span class="va">Replicate</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">exprs_data</span><span class="op">)</span>

<span class="va">pep_res</span>
<span class="co">#&gt; MSnSet (storageMode: lockedEnvironment)</span>
<span class="co">#&gt; assayData: 454 features, 4 samples </span>
<span class="co">#&gt;   element names: exprs </span>
<span class="co">#&gt; protocolData: none</span>
<span class="co">#&gt; phenoData</span>
<span class="co">#&gt;   sampleNames: ratio_1 ratio_2 ratio_4 ratio_3</span>
<span class="co">#&gt;   varLabels: Replicate</span>
<span class="co">#&gt;   varMetadata: labelDescription</span>
<span class="co">#&gt; featureData</span>
<span class="co">#&gt;   featureNames: AAAAAAAAAAAAAAAGAGAGAK_ AAAAAAALQAK_ ... YVSSLTEEISK_</span>
<span class="co">#&gt;     (454 total)</span>
<span class="co">#&gt;   fvarLabels: Master.Protein.Accessions Sequence ... Matched_NC_3 (7</span>
<span class="co">#&gt;     total)</span>
<span class="co">#&gt;   fvarMetadata: labelDescription</span>
<span class="co">#&gt; experimentData: use 'experimentData(object)'</span>
<span class="co">#&gt; Annotation:  </span>
<span class="co">#&gt; - - - Processing information - - -</span>
<span class="co">#&gt;  MSnbase version: 2.18.0</span></code></pre></div>
<p>Then we summarise the peptide-level rations to protein-level ratios. Here, we’ll use the median and demand at least two ratios per protein.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># see ?camprotR::restrict_features_per_protein for more details</span>
<span class="co"># In particular, note that the function will replace some quantification values </span>
<span class="co"># with NA, while removing some other peptides entirely</span>
<span class="va">pep_res_flt</span> <span class="op">&lt;-</span> <span class="fu">camprotR</span><span class="fu">::</span><span class="fu"><a href="../reference/restrict_features_per_protein.html">restrict_features_per_protein</a></span><span class="op">(</span>
  <span class="va">pep_res</span>, 
  min_features <span class="op">=</span> <span class="fl">2</span>, 
  plot <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>
<span class="co">#&gt; `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre></div>
<p><img src="SILAC_files/figure-html/unnamed-chunk-19-1.png" width="384"></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">prot_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/combineFeatures.html" class="external-link">combineFeatures</a></span><span class="op">(</span>
  <span class="va">pep_res_flt</span>, 
  groupBy <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">pep_res_flt</span><span class="op">)</span><span class="op">$</span><span class="va">Master.Protein.Accessions</span>,
  method <span class="op">=</span> <span class="st">"median"</span>,
  na.rm <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>
<span class="co">#&gt; Your data contains missing values. Please read the relevant section in</span>
<span class="co">#&gt; the combineFeatures manual page for details on the effects of missing</span>
<span class="co">#&gt; values on data aggregation.</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">pep_res</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] 454</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">pep_res_flt</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] 417</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="op">(</span><span class="va">prot_res</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] 59</span></code></pre></div>
<p>Note that the summarisation is occurring on the expression data, but will affect the feature data too (the values in the first feature are taken). If we want to retain information about whether the peptide was ‘matched’ in both conditions, we will need to generate this ourselves and re-attach to the MSnSet.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">prot_matched_nc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">pep_res_flt</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">Sequence</span>, <span class="op">-</span><span class="va">Modifications</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span>cols <span class="op">=</span> <span class="op">-</span><span class="va">Master.Protein.Accessions</span>, values_to <span class="op">=</span> <span class="st">'matched_NC'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">Master.Protein.Accessions</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">matched_NC</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>any_not_matched <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/any.html" class="external-link">any</a></span><span class="op">(</span><span class="op">!</span><span class="va">matched_NC</span><span class="op">)</span><span class="op">)</span>


<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">prot_res</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">prot_res</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"id"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">prot_matched_nc</span>, by <span class="op">=</span> <span class="st">"Master.Protein.Accessions"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"id"</span><span class="op">)</span></code></pre></div>
<p>We can now inspect the completeness of our protein-level data using <code><a href="https://lgatto.github.io/MSnbase/reference/plotNA-methods.html" class="external-link">MSnbase::plotNA()</a></code>. In this case 18/59 proteins have complete data (all 4 replicates) and the expression matrix is 63% complete overall.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">MSnbase</span><span class="fu">::</span><span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/plotNA-methods.html" class="external-link">plotNA</a></span><span class="op">(</span><span class="va">prot_res</span>, pNA <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></code></pre></div>
<p><img src="SILAC_files/figure-html/unnamed-chunk-21-1.png" width="384"></p>
<p>To identify proteins with significantly increased abundance in CL vs NC, we will use DEqMS <span class="citation">(Zhu et al. 2020)</span>, which you can think of as an extension of limma <span class="citation">(Ritchie et al. 2015)</span> specifically for proteomics.</p>
<p>The next step is taken from the <a href="https://bioconductor.org/packages/release/bioc/vignettes/DEqMS/inst/doc/DEqMS-package-vignette.html#extract-quant-data-columns-for-deqms" class="external-link">DEqMS vignette</a>. The idea is that we want to create an <code>MArrayLM</code> object as per normal <code>limma</code> analysis and then add a <code>$count</code> column to the <code>MArrayLM</code> object and then use the <code>spectraCounteBayes</code> function to perform the Bayesian shrinkage using the count column rather than the <code>$Amean</code> column. In our case, <code>count</code> is the number of peptides per protein. In the DEqMS paper, they suggest that the best summarisation metric to use is the minimum value across the samples.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dat</span> <span class="op">&lt;-</span> <span class="va">prot_res</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/MSnSet-class.html" class="external-link">filterNA</a></span><span class="op">(</span>pNA <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="co"># select proteins in min. 2/4 reps</span>
  <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="op">)</span>

<span class="va">design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span>Intercept <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">dat</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">lmFit</a></span><span class="op">(</span><span class="va">dat</span>, <span class="va">design</span><span class="op">)</span>
<span class="va">efit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span>

<span class="co"># Obtain the min peptide count across the samples</span>
<span class="va">min_pep_count</span> <span class="op">&lt;-</span> <span class="fu">camprotR</span><span class="fu">::</span><span class="fu"><a href="../reference/count_features_per_protein.html">count_features_per_protein</a></span><span class="op">(</span><span class="va">pep_res</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">Master.Protein.Accessions</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>min_pep_count <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span>

<span class="co"># add the min peptide count</span>
<span class="va">efit</span><span class="op">$</span><span class="va">count</span> <span class="op">&lt;-</span> <span class="va">min_pep_count</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Master.Protein.Accessions</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">efit</span><span class="op">$</span><span class="va">coefficients</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">min_pep_count</span><span class="op">)</span>

<span class="co"># run DEqMS</span>
<span class="va">efit_deqms</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/DEqMS/man/spectraCounteBayes.html" class="external-link">spectraCounteBayes</a></span><span class="op">(</span><span class="va">efit</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Unfortunately, since we have so few proteins in this toy dataset, the relationship between PSM count and variance is not apparent, though it should be with a full-scale proteomics experiment.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Diagnostic plots</span>
<span class="fu"><a href="https://rdrr.io/pkg/DEqMS/man/VarianceBoxplot.html" class="external-link">VarianceBoxplot</a></span><span class="op">(</span><span class="va">efit_deqms</span>, n <span class="op">=</span> <span class="fl">30</span>, xlab <span class="op">=</span> <span class="st">"PSM count"</span><span class="op">)</span></code></pre></div>
<p><img src="SILAC_files/figure-html/unnamed-chunk-23-1.png" width="384"></p>
<p>Below, we summarise the number of proteins with statistically different abundance in CL vs NC and plot a ‘volcano’ plot to visualise this. Note that all 24 proteins with a statistically significant change are increased in CL, so the plot looks more like a fire-hose than a volcano!</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">deqms_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DEqMS/man/outputResult.html" class="external-link">outputResult</a></span><span class="op">(</span><span class="va">efit_deqms</span>, coef_col <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">deqms_results</span><span class="op">$</span><span class="va">sca.adj.pval</span> <span class="op">&lt;</span> <span class="fl">0.01</span>, <span class="st">'sig.'</span>, <span class="st">'not sig.'</span><span class="op">)</span>,
      <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">deqms_results</span><span class="op">$</span><span class="va">logFC</span> <span class="op">&gt;</span> <span class="fl">1</span>, <span class="st">'Higher in CL'</span>, <span class="st">'Lower in CL'</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt;           </span>
<span class="co">#&gt;            Higher in CL Lower in CL</span>
<span class="co">#&gt;   not sig.            1          19</span>
<span class="co">#&gt;   sig.               24           0</span>

<span class="va">deqms_results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">logFC</span>, y <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">sca.P.Value</span><span class="op">)</span>, colour <span class="op">=</span> <span class="va">sca.adj.pval</span> <span class="op">&lt;</span> <span class="fl">0.01</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/theme_camprot.html">theme_camprot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_colour_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'grey'</span>, <span class="fu"><a href="../reference/get_cat_palette.html">get_cat_palette</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>, name <span class="op">=</span> <span class="st">'CL vs NC Sig.'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">'CL vs NC (Log2)'</span>, y <span class="op">=</span> <span class="st">'-log10(p-value)'</span><span class="op">)</span></code></pre></div>
<p><img src="SILAC_files/figure-html/unnamed-chunk-24-1.png" width="480"></p>
<p>Finally, we can explore and export out results. Importantly, the <code>$t</code>, <code>$P.Value</code> and <code>$adj.P.Val</code> columnts are from <code>limma</code>. The columns prefixed with <code>sca</code> are the from <code>DEqMS</code>.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">head</a></span><span class="op">(</span><span class="va">deqms_results</span><span class="op">)</span>
<span class="co">#&gt;           logFC  AveExpr         t      P.Value    adj.P.Val         B   gene</span>
<span class="co">#&gt; P15880 5.843093 5.843093 19.119852 1.889018e-33 8.311680e-32 174.55385 P15880</span>
<span class="co">#&gt; Q92945 5.688033 5.688033 18.612464 1.310927e-32 2.884040e-31 165.03692 Q92945</span>
<span class="co">#&gt; P36578 3.775106 3.775106 12.352958 4.392638e-21 3.865522e-20  68.68522 P36578</span>
<span class="co">#&gt; P40429 3.126439 3.126439  8.859768 6.740128e-14 3.295174e-13  31.91773 P40429</span>
<span class="co">#&gt; P61254 4.865011 4.865011 11.256688 7.293579e-19 5.348624e-18  55.79930 P61254</span>
<span class="co">#&gt; O43390 5.876200 5.876200 16.652098 3.151435e-29 4.622105e-28 130.54854 O43390</span>
<span class="co">#&gt;        count    sca.t  sca.P.Value sca.adj.pval</span>
<span class="co">#&gt; P15880     4 39.45864 1.101209e-37 4.845318e-36</span>
<span class="co">#&gt; Q92945     5 29.40900 6.460325e-32 1.421272e-30</span>
<span class="co">#&gt; P36578     3 20.64553 3.275304e-25 4.803779e-24</span>
<span class="co">#&gt; P40429     1 16.21849 1.187789e-20 1.306568e-19</span>
<span class="co">#&gt; P61254     1 14.41418 1.776341e-18 1.563180e-17</span>
<span class="co">#&gt; O43390     2 12.80037 9.098531e-17 6.672256e-16</span></code></pre></div>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-http://zotero.org/users/5634351/items/6HPNEJ3F" class="csl-entry">
Krüger, Marcus, Markus Moser, Siegfried Ussar, Ingo Thievessen, Christian A. Luber, Francesca Forner, Sarah Schmidt, Sara Zanivan, Reinhard Fässler, and Matthias Mann. 2008. <span>“SILAC Mouse for Quantitative Proteomics Uncovers Kindlin-3 as an Essential Factor for Red Blood Cell Function.”</span> <em>Cell</em> 134 (2): 353–64. <a href="https://doi.org/10.1016/j.cell.2008.05.033" class="external-link">https://doi.org/10.1016/j.cell.2008.05.033</a>.
</div>
<div id="ref-http://zotero.org/users/5634351/items/ST368TPT" class="csl-entry">
Ong, Shao-En, Blagoy Blagoev, Irina Kratchmarova, Dan Bach Kristensen, Hanno Steen, Akhilesh Pandey, and Matthias Mann. 2002. <span>“Stable Isotope Labeling by Amino Acids in Cell Culture, SILAC, as a Simple and Accurate Approach to Expression Proteomics*.”</span> <em>Molecular &amp; Cellular Proteomics</em> 1 (5): 376–86. <a href="https://doi.org/10.1074/mcp.M200025-MCP200" class="external-link">https://doi.org/10.1074/mcp.M200025-MCP200</a>.
</div>
<div id="ref-http://zotero.org/users/5634351/items/EJHWF46N" class="csl-entry">
Queiroz, Rayner M. L., Tom Smith, Eneko Villanueva, Maria Marti-Solano, Mie Monti, Mariavittoria Pizzinga, Dan-Mircea Mirea, et al. 2019. <span>“Comprehensive identification of RNA–protein interactions in any organism using orthogonal organic phase separation (OOPS).”</span> <em>Nature Biotechnology</em> 37 (2): 169. <a href="https://doi.org/10.1038/s41587-018-0001-2" class="external-link">https://doi.org/10.1038/s41587-018-0001-2</a>.
</div>
<div id="ref-http://zotero.org/users/5634351/items/6KTXTWME" class="csl-entry">
Ritchie, Matthew E., Belinda Phipson, Di Wu, Yifang Hu, Charity W. Law, Wei Shi, and Gordon K. Smyth. 2015. <span>“Limma Powers Differential Expression Analyses for RNA-Sequencing and Microarray Studies.”</span> <em>Nucleic Acids Research</em> 43 (7): e47. <a href="https://doi.org/10.1093/nar/gkv007" class="external-link">https://doi.org/10.1093/nar/gkv007</a>.
</div>
<div id="ref-http://zotero.org/users/5634351/items/RTM6NFVU" class="csl-entry">
Zhu, Yafeng, Lukas M. Orre, Yan Zhou Tran, Georgios Mermelekas, Henrik J. Johansson, Alina Malyutina, Simon Anders, and Janne Lehtiö. 2020. <span>“DEqMS: A Method for Accurate Variance Estimation in Differential Protein Expression Analysis.”</span> <em>Molecular &amp; Cellular Proteomics : MCP</em> 19 (6): 1047–57. <a href="https://doi.org/10.1074/mcp.TIR119.001646" class="external-link">https://doi.org/10.1074/mcp.TIR119.001646</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Charlotte Dawson, Tom Smith.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
